diff --git a/PKGBUILD b/PKGBUILD
index 8ac9488..c5c0fe8 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -4,18 +4,19 @@ pkgbase=linux-tools
 pkgname=(
   'bootconfig'
   'bpf'
+#  'cgroup_event_listener'
   'cpupower'
   'hyperv'
   'linux-tools-meta'
   'perf'
   'tmon'
-  'turbostat'
+#  'turbostat'
   'usbip'
-  'x86_energy_perf_policy'
+#  'x86_energy_perf_policy'
 )
-pkgver=6.10
-pkgrel=1
-license=('GPL-2.0-only')
+pkgver=6.8
+pkgrel=
+license=('GPL2')
 arch=('x86_64')
 url='https://www.kernel.org'
 options=('!strip' '!lto')
@@ -40,11 +41,12 @@ makedepends+=('libcap')
 makedepends+=('llvm' 'clang')
 groups=("$pkgbase")
 source=("git+https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git#tag=v${pkgver//_/-}?signed"
-#        "https://cdn.kernel.org/pub/linux/kernel/v6.x/patch-$pkgver.2.xz"
+        #"https://cdn.kernel.org/pub/linux/kernel/v6.x/patch-$pkgver.7.xz"
         'cpupower.default'
         'cpupower.systemd'
         'cpupower.service'
         'usbipd.service'
+        'hv_fcopy_daemon.service'
         'hv_kvp_daemon.service'
         'hv_vss_daemon.service'
 )
@@ -52,15 +54,17 @@ validpgpkeys=(
   'ABAF11C65A2970B130ABE3C479BE3E4300411886'  # Linus Torvalds
   '647F28654894E3BD457199BE38DBBDC86092693E'  # Greg Kroah-Hartman
 )
-sha256sums=('e7ef67dfdf83e23ca49bb679dd32782c6d06088696bfa3b3a626705f51635d29'
+sha256sums=('SKIP'
             '4fa509949d6863d001075fa3e8671eff2599c046d20c98bb4a70778595cd1c3f'
             'b692f4859ed3fd9831a058a450a84d8c409bf7e3e45aac1c2896a14bb83f3d7a'
             '42d2ec9f1d9cc255ee7945a27301478364ef482f5a6ddfc960189f03725ccec2'
             '2e187734d8aec58a3046d79883510d779aa93fb3ab20bd3132c1a607ebe5498f'
+            '16855c197d2334f820cb190312a5a7fffe9165189db01344a957e582e39e17d8'
             'b1315cb77a35454e1af9172f821a52e2a0cb18561be05a340d21cf337b01ae61'
             '2d5e2f8d40b6f19bf2e1dead57ca105d72098fb0b418c09ff2e0cb91089710af')
 
 prepare() {
+  ln -sf linux-$pkgver linux
   cd linux
 
   # apply patch from the source array (should be a pacman feature)
@@ -87,7 +91,7 @@ build() {
     prefix=/usr \
     lib=lib/perf \
     perfexecdir=lib/perf \
-    WERROR=0 \
+    EXTRA_CFLAGS=' -Wno-error=bad-function-cast -Wno-error=declaration-after-statement -Wno-error=switch-enum' \
     NO_SDT=1 \
     BUILD_BPF_SKEL=1 \
     PYTHON=python \
@@ -100,10 +104,10 @@ build() {
   make VERSION=$pkgver-$pkgrel
   popd
 
-  echo ':: x86_energy_perf_policy'
-  pushd linux/tools/power/x86/x86_energy_perf_policy
-  make
-  popd
+#  echo ':: x86_energy_perf_policy'
+#  pushd linux/tools/power/x86/x86_energy_perf_policy
+#  make
+#  popd
 
   echo ':: usbip'
   pushd linux/tools/usb/usbip
@@ -119,14 +123,19 @@ build() {
   make
   popd
 
-  echo ':: turbostat'
-  pushd linux/tools/power/x86/turbostat
-  make
-  popd
+#  echo ':: cgroup_event_listener'
+#  pushd linux/tools/cgroup
+#  make
+#  popd
+
+#  echo ':: turbostat'
+#  pushd linux/tools/power/x86/turbostat
+#  make
+#  popd
 
   echo ':: hv'
   pushd linux/tools/hv
-  CFLAGS+=' -DKVP_SCRIPTS_PATH=\"/usr/lib/hypervkvpd/\"' make
+  CFLAGS+=' -DKVP_SCRIPTS_PATH=\"/usr/lib/hyperv/kvp_scripts/\"' make
   popd
 
   echo ':: bpf'
@@ -150,13 +159,14 @@ package_linux-tools-meta() {
   depends=(
     'bootconfig'
     'bpf'
+#    'cgroup_event_listener'
     'cpupower'
     'hyperv'
     'perf'
     'tmon'
-    'turbostat'
+#    'turbostat'
     'usbip'
-    'x86_energy_perf_policy'
+#    'x86_energy_perf_policy'
   )
   conflicts=(
     'acpidump'
@@ -246,6 +256,14 @@ package_tmon() {
   make install INSTALL_ROOT="$pkgdir"
 }
 
+package_cgroup_event_listener() {
+  pkgdesc='Simple listener of cgroup events'
+  depends=('glibc')
+
+  cd linux/tools/cgroup
+  install -Dm755 cgroup_event_listener "$pkgdir/usr/bin/cgroup_event_listener"
+}
+
 package_turbostat() {
   pkgdesc='Report processor frequency and idle statistics'
   depends=('glibc' 'libcap')
@@ -259,10 +277,11 @@ package_hyperv() {
   depends=('glibc')
 
   cd linux/tools/hv
-  make install DESTDIR="$pkgdir" sbindir=/usr/bin libexecdir=/usr/lib
-  for _p in hv_kvp_daemon hv_vss_daemon; do
+  for _p in hv_fcopy_daemon hv_kvp_daemon hv_vss_daemon; do
+    install -Dm755 "$_p" "$pkgdir/usr/bin/$_p"
     install -Dm644 "$srcdir/$_p.service" "$pkgdir/usr/lib/systemd/system/$_p.service"
   done
+  install -dm755 "$pkgdir/usr/lib/hyperv/kvp_scripts"
 }
 
 package_bpf() {
