diff --git a/PKGBUILD b/PKGBUILD
index c5fefa4..49cb018 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -42,7 +42,7 @@ prepare() {
   # NOTE: we are renaming hx to helix so there is no conflict with hex (providing hx)
   sed -i "s|hx|helix|g" contrib/completion/hx.*
   sed -i 's|hx|helix|g' contrib/Helix.desktop
-  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+  cargo fetch --locked --target "`uname -m`-unknown-linux-gnu"
 }
 
 build() {
