diff --git a/PKGBUILD b/PKGBUILD
index d0d3da8..3f1b55b 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -141,7 +141,7 @@ package_ruby() {
     rm --force --recursive --verbose \
       "${pkgdir}"/usr/lib/ruby/${rubyver}/${stdlib_gem} \
       "${pkgdir}"/usr/lib/ruby/${rubyver}/${stdlib_gem}.rb \
-      "${pkgdir}"/usr/lib/ruby/${rubyver}/x86_64-linux/${stdlib_gem}.so \
+      "${pkgdir}"/usr/lib/ruby/${rubyver}/`uname -m`-linux/${stdlib_gem}.so \
       "${pkgdir}"/usr/lib/ruby/gems/${rubyver}/specifications/default/${stdlib_gem}-*.gemspec
   done
 
@@ -151,14 +151,14 @@ package_ruby() {
     "${pkgdir}"/usr/lib/ruby/${rubyver}/net/http \
     "${pkgdir}"/usr/lib/ruby/${rubyver}/net/http.rb \
     "${pkgdir}"/usr/lib/ruby/${rubyver}/net/https.rb \
-    "${pkgdir}"/usr/lib/ruby/${rubyver}/x86_64-linux/cgi \
-    "${pkgdir}"/usr/lib/ruby/${rubyver}/x86_64-linux/date_core.so \
-    "${pkgdir}"/usr/lib/ruby/${rubyver}/x86_64-linux/digest \
-    "${pkgdir}"/usr/lib/ruby/${rubyver}/x86_64-linux/io/console.so \
-    "${pkgdir}"/usr/lib/ruby/${rubyver}/x86_64-linux/io/nonblock.so \
-    "${pkgdir}"/usr/lib/ruby/${rubyver}/x86_64-linux/io/wait.so \
-    "${pkgdir}"/usr/lib/ruby/${rubyver}/x86_64-linux/json \
-    "${pkgdir}"/usr/lib/ruby/${rubyver}/x86_64-linux/racc
+    "${pkgdir}"/usr/lib/ruby/${rubyver}/`uname -m`-linux/cgi \
+    "${pkgdir}"/usr/lib/ruby/${rubyver}/`uname -m`-linux/date_core.so \
+    "${pkgdir}"/usr/lib/ruby/${rubyver}/`uname -m`-linux/digest \
+    "${pkgdir}"/usr/lib/ruby/${rubyver}/`uname -m`-linux/io/console.so \
+    "${pkgdir}"/usr/lib/ruby/${rubyver}/`uname -m`-linux/io/nonblock.so \
+    "${pkgdir}"/usr/lib/ruby/${rubyver}/`uname -m`-linux/io/wait.so \
+    "${pkgdir}"/usr/lib/ruby/${rubyver}/`uname -m`-linux/json \
+    "${pkgdir}"/usr/lib/ruby/${rubyver}/`uname -m`-linux/racc
 }
 
 package_ruby-docs() {
