diff --git a/PKGBUILD b/PKGBUILD
index 8d075a7..a37e056 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -38,7 +38,7 @@ build() {
     CFLAGS="$CFLAGS -fPIC -Wl,-z,relro,-z,now -w -I/usr/include/mbedtls2" \
     CXXFLAGS="$CXXFLAGS -fPIC -Wl,-z,relro,-z,now -w -I/usr/include/mbedtls2" \
     LINKFLAGS="$LDFLAGS -L/usr/lib/mbedtls2" \
-    arch=$CARCH \
+    arch=`uname -m` \
     builtin_embree=no \
     builtin_enet=yes \
     builtin_freetype=no \
@@ -82,5 +82,5 @@ package() {
   install -Dm644 misc/dist/linux/godot.6 "$pkgdir/usr/share/man/man6/godot.6"
   install -Dm644 misc/dist/linux/org.godotengine.Godot.xml \
     "$pkgdir/usr/share/mime/packages/org.godotengine.Godot.xml"
-  install -Dm755 bin/godot.linuxbsd.editor.$CARCH "$pkgdir/usr/bin/godot"
+  install -Dm755 bin/godot.linuxbsd.editor.`uname -m` "$pkgdir/usr/bin/godot"
 }
