diff --git a/PKGBUILD b/PKGBUILD
index 154b187..b4f1e4a 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -11,7 +11,7 @@ pkgrel=1
 pkgdesc='A secure web-based, collaborative terminal'
 url="https://github.com/ekzhang/$pkgname"
 license=(MIT)
-arch=(x86_64 aarch64)
+arch=(loong64 x86_64 aarch64)
 depends=(gcc-libs
          glibc)
 makedepends=(cargo
@@ -23,7 +23,7 @@ sha256sums=('5773c2c65dea72ec4b98d22b0c0534b745c2c536ea5b1267ced5c668b3a6736f')
 
 prepare() {
 	cd "$_archive"
-	cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+	cargo fetch --locked --target "`uname -m`-unknown-linux-gnu"
 }
 
 build() {
