diff --git a/PKGBUILD b/PKGBUILD
index 83235f9..507d49d 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -14,7 +14,7 @@ arch=('x86_64')
 depends=('bzip2' 'libltdl' 'libxml2' 'curl' 'systemd-libs' 'pcre2' 'json-c' 'libmspack'
          'ncurses' libsystemd.so libncursesw.so libcurl.so libjson-c.so libbz2.so)
 makedepends=('libmilter' 'systemd' 'cmake' 'ninja' 'python' 'cargo')
-checkdepends=('check')
+makedepends+=('check')
 backup=('etc/clamav/clamd.conf'
         'etc/clamav/freshclam.conf'
         'etc/clamav/clamav-milter.conf'
