diff --git a/PKGBUILD b/PKGBUILD
index b633908..5f263b3 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -13,12 +13,15 @@ depends=('qt5-base')
 makedepends=('git')
 groups=('qt5')
 _pkgfqn=${pkgname/5-/}
-source=(git+https://code.qt.io/qt/$_pkgfqn.git#tag=v${pkgver}-lts)
-sha256sums=('SKIP')
+source=(git+https://code.qt.io/qt/$_pkgfqn.git#tag=v${pkgver}-lts
+        loongarch_ports.patch)
+sha256sums=('SKIP'
+            'e6b6bd257375ba0982f6be8cd9c704a8cf3101d275ee1c40cad8466c695d3800')
 options=(!lto)
 
 prepare() {
   mkdir -p build
+  patch -p1 -d qtscript/src -i "$srcdir"/loongarch_ports.patch
 }
 
 build() {
