diff --git a/PKGBUILD b/PKGBUILD
index 34fab60..62e3644 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -24,7 +24,7 @@ build() {
   )
 
   arch-meson "${pkgname}-${pkgver}" build "${_meson_options[@]}"
-
+#  arch-meson -D rulesdir="/usr/lib/udev/rules.d build" -Discsi_sbindir=/usr/bin build
   meson compile -C build
 }
 
@@ -33,6 +33,7 @@ package() {
 
   # no initiatorname in package, initialized in install script
   echo -n > "${pkgdir}"/etc/iscsi/initiatorname.iscsi
+#  meson install -C build --destdir "$pkgdir"
 
   # install docs
   install -D -m0644 -t "${pkgdir}"/usr/share/doc/${pkgname}/ "${pkgname}-${pkgver}"/{Changelog,README}
