diff --git a/PKGBUILD b/PKGBUILD
index 198b2bc..296245c 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -12,7 +12,7 @@ arch=(x86_64)
 url="https://www.hdfgroup.org/hdf5"
 license=(custom)
 depends=(zlib libaec bash)
-makedepends=(cmake time gcc-fortran java-environment)
+makedepends=(cmake time gcc-fortran) # java-environment)
 replaces=(hdf5-java)
 provides=(hdf5-java)
 source=(https://support.hdfgroup.org/ftp/HDF5/releases/${pkgname}-${pkgver:0:4}/${pkgname}-${pkgver/_/-}/src/${pkgname}-${pkgver/_/-}.tar.bz2)
@@ -35,7 +35,7 @@ build() {
         -DHDF5_BUILD_HL_LIB=ON \
         -DHDF5_BUILD_CPP_LIB=ON \
         -DHDF5_BUILD_FORTRAN=ON \
-        -DHDF5_BUILD_JAVA=ON \
+        -DHDF5_BUILD_JAVA=OFF \
         -DHDF5_ENABLE_Z_LIB_SUPPORT=ON \
         -DHDF5_ENABLE_SZIP_SUPPORT=ON \
         -DHDF5_ENABLE_SZIP_ENCODING=ON \
@@ -52,7 +52,7 @@ build() {
         --enable-hl \
         --enable-cxx \
         --enable-fortran \
-        --enable-java \
+        --disable-java \
         --with-pic \
         --with-zlib \
         --with-szlib
