diff --git a/PKGBUILD b/PKGBUILD
index f52a11a..109ca9f 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -3,8 +3,8 @@
 # Contributor: Douglas Soares de Andrade <douglas@archlinux.org>
 
 pkgname=avahi
-pkgver=0.8+r194+g3f79789
-pkgrel=2
+pkgver=0.8+r194+g3f79789c
+pkgrel=
 epoch=1
 pkgdesc="Service Discovery for Linux using mDNS/DNS-SD (compatible with Bonjour)"
 url="https://github.com/avahi/avahi"
@@ -21,7 +21,6 @@ depends=(
 makedepends=(
   doxygen
   git
-  glib2-devel
   gobject-introspection
   graphviz
   gtk3
