diff --git a/PKGBUILD b/PKGBUILD
index 51acaaf..4257131 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -11,7 +11,7 @@ url="https://orc.apache.org"
 license=(Apache)
 depends=(lz4 protobuf snappy zlib zstd)
 makedepends=(cmake)
-checkdepends=(gtest)
+makedepends+=(gtest)
 source=(https://dlcdn.apache.org/${_pkg}/${_pkg}-${pkgver}/${_pkg}-${pkgver}.tar.gz{,.asc})
 sha256sums=('0dca8bbccdb2ee87e59ba964933436beebd02ea78c4134424828a8127fbc4faa'
             'SKIP')
@@ -40,7 +40,9 @@ build(){
     -DORC_PREFER_STATIC_ZLIB=OFF \
     -DBUILD_LIBHDFSPP=OFF \
     -DBUILD_JAVA=OFF \
-    -DINSTALL_VENDORED_LIBS=OFF
+    -DSTOP_BUILD_ON_WARNING=OFF \
+    -DINSTALL_VENDORED_LIBS=OFF \
+    -DBUILD_CPP_TESTS=OFF
   make -C build
 }
 
