diff --git a/PKGBUILD b/PKGBUILD
index b7adbee..1f371ac 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -42,15 +42,19 @@ provides=(python-archinstall)
 source=(
   $pkgname-v$pkgver.tar.gz::$url/archive/refs/tags/v$pkgver.tar.gz
   $pkgname-v$pkgver.tar.gz.sig::$url/releases/download/v$pkgver/$pkgname-$pkgver.tar.gz.sig
+  archinstall-la64-2.6.3.patch
 )
 sha512sums=('b88a301ff747f10f5b2e23b8c9217f28b54a5846123580eba06d1960a64e56357ec85414f3c0ce8b641cc2038fd5244608bdbac3ee00bef0a55928a51add0c05'
-            'SKIP')
+            'SKIP'
+            'bba2c25733ab266f2b848f2e569dc6b03a0dd9e844cdb976a16fe0190145c6d7858b6530fd0ac9785579968f43fba0e9037802b62d160b98832076e09f5376df')
 b2sums=('105b297c649e08edb67e36f1675a65c4b9a930cee154e7bdf9cde8d0a5af25ca8045f31fe2cdcccc076dc278dd9f952fbc03fe57f7436118e623c96cddcc8a4a'
-        'SKIP')
+        'SKIP'
+        'ed1717d1f5649383bc15db231636181a135857da3ec4f51d2e559b7af402c4013ba6e0daa0d5af154ca00f3fea15a88c11e97b8c9060a867345cb361b2b8151b')
 validpgpkeys=('8AA2213C8464C82D879C8127D4B58E897A929F2E') # torxed@archlinux.org
 
 prepare() {
   cd $pkgname-$pkgver
+  patch -p1 -i $srcdir/archinstall-la64-2.6.3.patch
 }
 
 build() {
