diff --git a/PKGBUILD b/PKGBUILD
index 633752e..525f1b3 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -30,7 +30,7 @@ build() {
   local _extension_api_version="$(ruby -e 'puts Gem.extension_api_version')"
 
   install --verbose --directory --mode=0755 \
-    "tmp_install_default/gemspec/specifications/gems/${_gemname}-${pkgver}" \
+     "tmp_install_default/gemspec/specifications/gems/${_gemname}-${pkgver}" \
     "tmp_install/usr/lib/ruby/${_extension_api_version}/${_platform}" \
     "tmp_install${_gemdir}/specifications/default"
 
