diff --git a/PKGBUILD b/PKGBUILD
index 84ed72e..4ffd82c 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -18,7 +18,7 @@ build() {
   export CGO_CFLAGS="${CFLAGS}"
   export CGO_CXXFLAGS="${CXXFLAGS}"
   export CGO_LDFLAGS="${LDFLAGS}"
-  export GOFLAGS="-buildmode=pie -trimpath -mod=readonly -modcacherw -ldflags=-linkmode=external"
+  export GOFLAGS="-trimpath -mod=readonly -modcacherw -ldflags=-linkmode=external"
   go build -v ./cmd/yaegi
 }
 
