diff --git a/PKGBUILD b/PKGBUILD
index 8457d84..95758f5 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -25,7 +25,7 @@ package() {
     _extramodules="/usr/lib/modules/${_kernver}/extramodules"
 
     install -Dm644 -t "${pkgdir}${_extramodules}" \
-        ${_module}/${pkgver}/${_kernver}/${CARCH}/module/*
+        ${_module}/${pkgver}/${_kernver}/`uname -m`/module/*
 
     # compress kernel modules
     find "$pkgdir" -name "*.ko" -exec xz {} +
