diff --git a/PKGBUILD b/PKGBUILD
index 744087e..6e2a773 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -5,7 +5,7 @@ pkgname=python-keyutils
 pkgver=0.6
 pkgrel=9
 pkgdesc="A set of python bindings for keyutils"
-arch=("x86_64")
+arch=("loong64" "x86_64")
 url="https://github.com/sassoftware/python-keyutils"
 license=("Apache")
 # python-pytest-runner in makedepends due to be set in setup_requires (which is wrong upstream)
