diff --git a/PKGBUILD b/PKGBUILD
index 59620fc..142c55c 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -11,8 +11,15 @@ depends=('gcc-libs')
 makedepends=('cmake' 'doctest' 'doxygen' 'ninja')
 # We're going to this alternate fork until the patches are upstreamed.
 # See https://github.com/erincatto/box2d/issues/621
-source=("$pkgname-$pkgver.tar.gz::https://github.com/erincatto/Box2D/archive/v${pkgver}.tar.gz")
-sha512sums=('d900f925b77906777719c91488bdc5e2df1ad1f4a8ca39a574229f5e57070e3a843bdd7530e817112605fde6d82145c872d8afdfc65b84531a73199098c81162')
+source=("$pkgname-$pkgver.tar.gz::https://github.com/erincatto/Box2D/archive/v${pkgver}.tar.gz"
+  box2d-fix-build.patch)
+sha512sums=('d900f925b77906777719c91488bdc5e2df1ad1f4a8ca39a574229f5e57070e3a843bdd7530e817112605fde6d82145c872d8afdfc65b84531a73199098c81162'
+            '74055d49f0b9f601c2e68576aa3e0ef43c061beb428cda3de847d5b9fb8bf6adb74b69521264f9aea048d268e5104be6bafdcfb0cfb09aec1de1662d263235e9')
+
+prepare() {
+  cd $pkgname-$pkgver
+  patch -p1 -i $srcdir/box2d-fix-build.patch
+}
 
 prepare() {
   # Use system doctest
