diff --git a/PKGBUILD b/PKGBUILD
index bbc0582..e2a8855 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -26,8 +26,10 @@ optdepends=(
   'openssl: for CMS signature verification of rootfs over PXE'
   'pv: for status display during copy to RAM'
 )
-source=(git+https://gitlab.archlinux.org/mkinitcpio/$pkgname.git#tag=$_commit?signed)
-sha256sums=('SKIP')
+source=(git+https://gitlab.archlinux.org/mkinitcpio/$pkgname.git#tag=$_commit?signed
+        mkinitcpio-archiso-loong64.patch)
+sha256sums=('SKIP'
+            '19bc8a202512b07c5600e599ddde8a61efe1b60b25c53162c77eed13578b6502')
 validpgpkeys=(
   '991F6E3F0765CF6295888586139B09DA5BF0D338'  # David Runge <dvzrv@archlinux.org>
   'BB8E6F1B81CF0BB301D74D1CBF425A01E68B38EF'  # nl6720 <nl6720@archlinux.org>
@@ -38,6 +40,11 @@ pkgver() {
   git describe --tags | sed 's/\([^-]*-g\)/r\1/;s/-/./g;s/v//g'
 }
 
+prepare() {
+  cd $pkgname
+  patch -p1 -i $srcdir/mkinitcpio-archiso-loong64.patch
+}
+
 check() {
   make -k check -C $pkgname
 }
