diff --git a/PKGBUILD b/PKGBUILD
index 970ca2b..0399644 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -6,7 +6,7 @@ pkgname=gocr
 pkgver=0.52
 pkgrel=2
 pkgdesc="OCR (Optical Character Recognition) program, which converts scanned images of text back to text files"
-arch=("x86_64")
+arch=("loong64" "x86_64")
 url="https://www-e.uni-magdeburg.de/jschulen/ocr/"
 license=('GPL')
 depends=('libjpeg-turbo' 'netpbm')
