diff --git a/PKGBUILD b/PKGBUILD
index 335b84a..39c83b5 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -5,7 +5,7 @@ pkgname=jgmenu
 pkgver=4.4.1
 pkgrel=1
 pkgdesc="Simple, independent, contemporary-looking X11 menu, designed for scripting, ricing and tweaking"
-arch=('i686' 'x86_64')
+arch=('i686' 'loong64' 'x86_64')
 url="https://www.github.com/johanmalm/jgmenu"
 license=('GPL')
 depends=('libx11' 'cairo' 'pango' 'libxrandr' 'librsvg' 'libxml2' 'glib2' 'menu-cache' 'python')
