diff --git a/PKGBUILD b/PKGBUILD
index 7e436da..aaedb1c 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -40,6 +40,8 @@ build() {
   export CGO_CPPFLAGS="${CPPFLAGS}"
   export CGO_CFLAGS="${CFLAGS}"
   export CGO_CXXFLAGS="${CXXFLAGS}"
+  go mod edit -replace=go.etcd.io/bbolt=go.etcd.io/bbolt@v1.3.8
+  go mod tidy
 
   local GIT_COMMIT="$(git rev-list -1 HEAD)"
   local GIT_LAST_TAG="$(git describe --abbrev=0 --tags)"
