diff --git a/PKGBUILD b/PKGBUILD
index db9e3d2..428c613 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -70,6 +70,10 @@ build() {
   export CGO_CFLAGS="${CFLAGS}"
   export CGO_CXXFLAGS="${CXXFLAGS}"
   export GOPATH="${srcdir}"
+  export GOPROXY=https://goproxy.cn
+  go mod edit -replace=golang.org/x/sys=github.com/golang/sys@v0.0.0-20220622161953-175b2fd9d664
+  go mod edit -replace=golang.org/x/net=github.com/golang/net@v0.0.0-20220622184535-263ec571b305
+  go mod tidy
 
   # build server/agent/cli
   go build -v \
