diff --git a/PKGBUILD b/PKGBUILD
index e7a06df..a6351f2 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -21,7 +21,7 @@ makedepends=(lua
 options=(debug)
 _archive="$_project-$_compatrel"
 _rockspec="$_rockname-${pkgver%_*}-$_rockrel.rockspec"
-_rock="$_rockname-${pkgver%_*}-$_rockrel.linux-$CARCH.rock"
+_rock="$_rockname-${pkgver%_*}-$_rockrel.linux-`uname -m`.rock"
 source=("$url/archive/v$_compatrel/$_archive.tar.gz"
         "${url/github/raw.githubusercontent}/1e31f3ddc517b4e521c73f7d6eaecd5e1787daa6/rockspecs/$_rockspec")
 sha256sums=('d1ed32f091856f6fffab06232da79c48b437afd4cd89e5c1fc85d7905b011430'
