diff --git a/PKGBUILD b/PKGBUILD
index 1abfbd1..7337e2b 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -22,7 +22,8 @@ source=(https://ftp.postgresql.org/pub/source/v${pkgver}/postgresql-${pkgver}.ta
         postgresql.sysusers
         postgresql.tmpfiles
         libxml2-2.12.patch
-        openssl3.2.patch)
+        openssl3.2.patch
+        add-loongarch-support.patch)
 md5sums=('9cbfb9076ed06384471802b850698a6d'
          '6ce1dab3da98a10f9190e6b3037f93aa'
          '632e22e96d6ace85b76a380487cfbf8c'
@@ -65,6 +66,7 @@ prepare() {
   patch -p1 < ../0002-Force-RPATH-to-be-used-for-the-PL-Perl-plugin.patch
   patch -p1 < ../libxml2-2.12.patch
   patch -p1 < ../openssl3.2.patch
+  patch -p1 < ../add-loongarch-support.patch
 }
 
 build() {
