diff --git a/PKGBUILD b/PKGBUILD
index a77a690..fb45226 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -29,13 +29,13 @@ sha512sums=('d871df84ebb3514b6474000b693246c8e5b198121e9a5a0ca18d478e54cd4144f8d
 validpgpkeys=('2CC8A0609AD2A479C65B6D5C8E8B898CBF2412F9')  # Weng Xuetian
 
 prepare() {
-  mkdir build
+  mkdir -p _build
 
   sed -e 's|enchant/enchant.h|enchant-2/enchant.h|' -i $pkgname-$pkgver/cmake/FindEnchant.cmake
 }
 
 build() {
-  cd build
+  cd _build
 
   cmake ../$pkgname-$pkgver \
     -DCMAKE_INSTALL_PREFIX=/usr \
@@ -53,11 +53,11 @@ build() {
 }
 
 check() {
-  cd build
+  cd _build
   make test
 }
 
 package() {
-  cd build
+  cd _build
   make DESTDIR="$pkgdir" install
 }
