diff --git a/PKGBUILD b/PKGBUILD
index 4a51f8a..439689e 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -31,7 +31,8 @@ build() {
 	export CGO_CFLAGS="${CFLAGS}"
 	export CGO_CXXFLAGS="${CXXFLAGS}"
 	export CGO_LDFLAGS="${LDFLAGS}"
-	export GOFLAGS="-buildmode=pie -trimpath -mod=readonly -modcacherw"
+    export GOPROXY=https://goproxy.cn
+	export GOFLAGS="-trimpath -mod=readonly -modcacherw"
 
 	#go build -o "$pkgname" .
 	make
