diff --git a/PKGBUILD b/PKGBUILD
index a68d4f8..337955d 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -10,11 +10,14 @@ pkgdesc='UI component responsible for handling the user interactions which can h
 arch=(x86_64)
 url='https://launchpad.net/online-accounts-signon-ui'
 license=(GPL)
-depends=(libaccounts-qt qt5-webengine signond libnotify)
+depends=(libaccounts-qt signond libnotify)
+makedepends=(qt5-declarative)
 source=(https://gitlab.com/accounts-sso/signon-ui/-/archive/$_commit/$pkgname-$pkgver.tar.gz
-        fake-user-agent.patch)
+        fake-user-agent.patch
+        signon-ui-loong64-bad-fix.patch)
 sha256sums=('4534b05d10e206d332bd91838af3411fdbfc897b604ba5843e7af76c1657f99b'
-            '5eb7782c6472e51a8107a25324d1d30052bac5d8e9050907cd957c89568fa577')
+            '5eb7782c6472e51a8107a25324d1d30052bac5d8e9050907cd957c89568fa577'
+            'ab79b1e4a67edeb6150e2cb592ec56c71c46a064a048cdb73ddd0533f96f09e0')
 options=(debug)
 
 prepare() {
@@ -24,6 +27,7 @@ prepare() {
   sed -e 's|src \\|src|' -e '/tests/d' -i signon-ui.pro
   # Fake user ID to bypass Google blacklist
   patch -p1 -i ../fake-user-agent.patch
+  patch -p1 -i $srcdir/signon-ui-loong64-bad-fix.patch
 }
 
 build() {
