diff --git a/PKGBUILD b/PKGBUILD
index 01d3ee1..b432bbf 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -14,8 +14,8 @@ license=('BSD')
 depends=("libmediainfo=$pkgver")
 makedepends=('wxwidgets-gtk3')
 source=("$pkgname-$pkgver.tar.gz::https://github.com/MediaArea/MediaInfo/archive/v$pkgver.tar.gz")
-sha512sums=('e5adb989bd9686d64c952794155993d7e7d9b0d81500e99cd9b7af61258bb71639ac1aa913dcd36ba51dca47e81ab8244de2bff37b603bd461cf13c42f59487a')
-b2sums=('9f0a1d1c0f9540be16963fb8f8d363f071efa25fcb2deb2494481d0eca58db79a29a6c5f8cc22eb01df30626ce44da7a0d921de24f6553e809e2588009445fa1')
+sha512sums=('33747cacd0657b67e7bc63596cb58cb4a8c6c2cab50c70b3075c1ffd12a9d165121a1dbcb2114a1e822273807341faa436181c3a0c3dfb1d98e45f120052d720')
+b2sums=('33b19a211b500caa0cb1af4ba17dfea70ffa1b169f7ff7a903614b53a08c54f6f8216f8a2e2bfffe02eef8733c741bc10af928111d336675634ce95418a5405d')
 
 build() {
   cd "MediaInfo-$pkgver"
