diff --git a/PKGBUILD b/PKGBUILD
index 59909d3..2a830f4 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -11,11 +11,11 @@ url="https://gitlab.com/accounts-sso/signon-plugin-oauth2"
 license=(LGPL)
 depends=(signond)
 makedepends=(qt5-xmlpatterns)
-source=("$pkgname-$pkgver.tar.gz::https://gitlab.com/accounts-sso/signon-plugin-oauth2/repository/archive.tar.gz?ref=VERSION_$pkgver")
-sha256sums=('f750a1e767c1977d73895b7aa7d5dca5494e4cbc53cf57c7e6c311933701ae52')
+source=("$pkgname-$pkgver.tar.gz::https://gitlab.com/accounts-sso/signon-plugin-oauth2/-/archive/VERSION_$pkgver/signon-plugin-oauth2-VERSION_$pkgver.tar.gz")
+sha256sums=('48a5bb61bf74fca458c870c1a27eedfe058b20f016c19a72aad2f5b151516447')
 
 prepare() {
-  cd signon-plugin-oauth2-VERSION_$pkgver-*
+  cd signon-plugin-oauth2-VERSION_$pkgver*
 
   sed -i '/-Werror/d' common-project-config.pri
 
@@ -25,14 +25,14 @@ prepare() {
 }
 
 build() {
-  cd signon-plugin-oauth2-VERSION_$pkgver-*
+  cd signon-plugin-oauth2-VERSION_$pkgver*
 
   qmake-qt5 PREFIX=/usr LIBDIR=/usr/lib
   make
 }
 
 package() {
-  cd signon-plugin-oauth2-VERSION_$pkgver-*
+  cd signon-plugin-oauth2-VERSION_$pkgver*
 
   make INSTALL_ROOT="$pkgdir" install
 }
