diff --git a/PKGBUILD b/PKGBUILD
index 686abfb..fdab826 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -20,7 +20,7 @@ build() {
   cd ninja-$pkgver
 
   python configure.py --bootstrap
-  emacs -Q --batch -f batch-byte-compile misc/ninja-mode.el
+  #emacs -Q --batch -f batch-byte-compile misc/ninja-mode.el
 }
 
 check() {
@@ -40,8 +40,8 @@ package() {
   install -m644 -D doc/manual.asciidoc "$pkgdir/usr/share/doc/ninja/manual.asciidoc"
   install -Dm644 COPYING "$pkgdir/usr/share/licenses/$pkgname/COPYING"
 
-  install -m644 -D misc/ninja-mode.el "$pkgdir/usr/share/emacs/site-lisp/ninja-mode.el"
-  install -m644 -D misc/ninja-mode.elc "$pkgdir/usr/share/emacs/site-lisp/ninja-mode.elc"
+  #install -m644 -D misc/ninja-mode.el "$pkgdir/usr/share/emacs/site-lisp/ninja-mode.el"
+  #install -m644 -D misc/ninja-mode.elc "$pkgdir/usr/share/emacs/site-lisp/ninja-mode.elc"
   install -m644 -D misc/ninja.vim "$pkgdir/usr/share/vim/vimfiles/syntax/ninja.vim"
   install -m644 -D misc/ninja_syntax.py "$pkgdir/$site_packages/ninja_syntax.py"
 
