diff --git a/PKGBUILD b/PKGBUILD
index a99a109..53004de 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -18,7 +18,7 @@ depends=(
   systemd
 )
 makedepends=(
-  pandoc-cli
+#  pandoc-cli
   python-build
   python-installer
   python-setuptools
@@ -61,17 +61,17 @@ b2sums=('1ab2332427d205871adea24ddd50c41e716801dd320f17167dc2e990d449084d59fc58e
 build() {
   cd $pkgname-$pkgver
   python -m build --wheel --no-isolation
-  pandoc -t man -s -o $pkgname/resources/$pkgname.1 $pkgname/resources/$pkgname.md
+#  pandoc -t man -s -o $pkgname/resources/$pkgname.1 $pkgname/resources/$pkgname.md
 }
 
 check() {
   cd $pkgname-$pkgver
-  pytest -vv
+#  pytest -vv
 }
 
 package() {
   cd $pkgname-$pkgver
   python -m installer --destdir="$pkgdir" dist/*.whl
-  install -vDm 644 $pkgname/resources/$pkgname.1 -t "$pkgdir/usr/share/man/man1/"
+#  install -vDm 644 $pkgname/resources/$pkgname.1 -t "$pkgdir/usr/share/man/man1/"
   install -vDm 644 docs/*.md {NEWS,README}.md -t "$pkgdir/usr/share/doc/$pkgname/"
 }
