diff --git a/PKGBUILD b/PKGBUILD
index d39a9fa..bdfd338 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -24,13 +24,19 @@ prepare() {
   cd "$pkgname"
 
   # download dependencies
-  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+  find -name Cargo.lock -exec rm -f {} \;
+  mkdir -p .cargo
+  cat >> .cargo/config <<EOF
+[source.crates-io]
+registry = "https://gitee.com/yetist/crates.io-index"
+EOF
+  cargo fetch --target "`uname -m`-unknown-linux-gnu"
 }
 
 build() {
   cd "$pkgname"
 
-  cargo build --frozen --release --all-features
+  cargo build --release --all-features
 }
 
 check() {
