diff --git a/PKGBUILD b/PKGBUILD
index 186ada5..72158f3 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -5,7 +5,7 @@ pkgname=zram-generator
 pkgver=1.1.2
 pkgrel=1
 pkgdesc="Systemd unit generator for zram devices"
-arch=("x86_64")
+arch=("loong64" "x86_64")
 url="https://github.com/systemd/zram-generator"
 license=('MIT')
 depends=("systemd")
@@ -15,6 +15,12 @@ sha256sums=('506d47acbabffa7013bb40a1f61c6edfa758a7bd55820d06ef49c7bc83dba762')
 
 build() {
   cd zram-generator-$pkgver
+  find -name Cargo.lock -exec rm -f {} \;
+  mkdir .cargo
+  cat > .cargo/config.toml <<EOF
+[source.crates-io]
+registry = "https://gitee.com/yetist/crates.io-index"
+EOF
   make CARGOFLAGS="--target-dir=target" build man
 }
 
