diff --git a/PKGBUILD b/PKGBUILD
index f11ef93..9746041 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -11,7 +11,7 @@ arch=(x86_64)
 url="https://ispc.github.io/"
 license=(BSD)
 depends=(clang gcc-libs glibc llvm-libs spirv-llvm-translator onetbb)
-makedepends=(cmake git level-zero-headers level-zero-loader lib32-glibc llvm openmp python vc-intrinsics)
+makedepends=(cmake git level-zero-headers level-zero-loader llvm openmp python vc-intrinsics)
 checkdepends=(intel-compute-runtime)
 optdepends=(
     'intel-compute-runtime: GPU support'
@@ -55,6 +55,11 @@ build() {
     -S $pkgname
     -W no-dev
   )
+# clang didn't support -mlsx
+  CFLAGS=${CFLAGS/-mlsx /}
+  CXXFLAGS=${CXXFLAGS/-mlsx /}
+  CFLAGS=${CFLAGS/-fstack-clash-protection/}
+  CXXFLAGS=${CXXFLAGS/-fstack-clash-protection/}
 
   cmake "${cmake_options[@]}"
   cmake --build build --verbose
