diff --git a/PKGBUILD b/PKGBUILD
index 85537fe..aad8dba 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -13,7 +13,8 @@ depends=('gcc-libs')
 makedepends=('libxml2' 'python' 'chrpath' 'doxygen' git cppunit help2man)
 optdepends=('python: tools')
 _commit=5b72244ca0d09c7f228d571ec2d5d20183486c11  # tags/v2.4.3
-source=("git+https://github.com/dell/libsmbios.git#commit=$_commit")
+source=("git+https://github.com/dell/libsmbios.git#commit=$_commit"
+0001-add-support-for-loongarch64.patch)
 sha256sums=('SKIP')
 
 pkgver() {
@@ -23,6 +24,7 @@ pkgver() {
 
 prepare() {
   cd $pkgname
+  patch -p1 -i $srcdir/0001-add-support-for-loongarch64.patch
   ./autogen.sh --no-configure
 }
 
@@ -48,3 +50,5 @@ package() {
   install -m755 -d "${pkgdir}/usr/share/licenses/${pkgname}"
   install -m644 ../COPYING-OSL "${pkgdir}/usr/share/licenses/${pkgname}/"
 }
+sha256sums=('SKIP'
+            '6bf9e8d30891867bd89698dbdb559a8c9d8c9878fd141a3ac1771993dfd6d420')
