diff --git a/PKGBUILD b/PKGBUILD
index d61d107..bddcb94 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -9,7 +9,7 @@ arch=('x86_64')
 url='https://nuspell.github.io/'
 license=('LGPL3')
 depends=('icu' 'glibc' 'gcc-libs')
-makedepends=('cmake' 'pandoc')
+makedepends=('cmake')
 checkdepends=('catch2')
 source=(${pkgname}-${pkgver}.tar.gz::"https://github.com/nuspell/nuspell/archive/v${pkgver}.tar.gz")
 sha512sums=('f4119b3fe5944be8f5bc35ccff8d7a93b0f4fa9f129bc97a7b96879a11b5b35bd714b41dd209267417e94c5fed45fd3a74b349f94424f4b90bde07d9694d1d7d')
