diff --git a/PKGBUILD b/PKGBUILD
index 957cc00..58a2926 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -5,7 +5,7 @@ pkgname=scantailor-advanced
 pkgver=1.0.19
 pkgrel=2
 pkgdesc="An interactive post-processing tool for scanned pages."
-arch=("x86_64")
+arch=("loong64" "x86_64")
 url="https://github.com/ScanTailor-Advanced/scantailor-advanced"
 license=("GPL3")
 depends=("qt6-base" "qt6-svg" "hicolor-icon-theme" "libpng>=1.0.0" "libtiff>=4.0.0")
