diff --git a/PKGBUILD b/PKGBUILD
index a2e8c2a..cf6f74e 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -9,7 +9,7 @@ arch=('x86_64')
 url="https://github.com/fcitx/fcitx5-m17n"
 license=('GPL')
 depends=('fcitx5' 'm17n-lib')
-makedepends=('extra-cmake-modules')
+makedepends=('extra-cmake-modules' 'fmt')
 source=("https://download.fcitx-im.org/fcitx5/$pkgname/$pkgname-$pkgver.tar.xz"{,.sig})
 sha512sums=('a28a170ba62edc9b6e225f5ff6d76c8d7333006429a6d496abd2b7b49a9f82e01a2ebfdd670609c96529bbee36a01fb1937abfab745f1d3f3a4f196f3b9fc3df'
             'SKIP')
