diff --git a/PKGBUILD b/PKGBUILD
index 011d63e..c66ae9a 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -9,7 +9,7 @@ arch=(x86_64)
 url='https://kde.org/plasma-desktop/'
 license=(LGPL)
 depends=(knewstuff5 kidletime5 qt5-graphicaleffects appstream-qt5 archlinux-appstream-data
-         kirigami2 discount kuserfeedback5 purpose5 qt5-webview)
+         kirigami2 discount kuserfeedback5 purpose5) # qt5-webview)
 makedepends=(extra-cmake-modules plasma-framework5 packagekit-qt5 flatpak fwupd)
 optdepends=('packagekit-qt5: to manage packages from Arch Linux repositories (not recommended, use at your own risk)'
             'flatpak: Flatpak packages support'
