diff --git a/PKGBUILD b/PKGBUILD
index 2309a19..bf50e22 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -10,7 +10,7 @@ pkgrel=1
 pkgdesc='GitLab Pages daemon used to serve static websites for GitLab users'
 url="https://gitlab.com/gitlab-org/$pkgname"
 license=(MIT)
-arch=(x86_64 i686)
+arch=(loong64 x86_64 i686)
 makedepends=(go)
 _archive="$pkgname-v$pkgver"
 source=("$url/-/archive/v$pkgver/$_archive.tar.bz2"
