diff --git a/PKGBUILD b/PKGBUILD
index 9f355c4..3941576 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -25,8 +25,8 @@ depends=(
   'systemd-libs'
 )
 makedepends=('libxslt' 'mongo-c-driver' 'librabbitmq-c' 'python' 'libesmtp' 'hiredis'
-             'libdbi' 'libmaxminddb' 'net-snmp' 'librdkafka' 'systemd')
-checkdepends=('python-nose' 'python-ply')
+             'libdbi' 'libmaxminddb' 'net-snmp' 'librdkafka' 'systemd') #'libcap'
+makedepends+=('python-nose' 'python-ply')
 optdepends=('logrotate: for rotating log files'
             'libdbi: for the SQL plugin'
             'librabbitmq-c: for the AMQP plugin'
