diff --git a/PKGBUILD b/PKGBUILD
index e5a027a..7bb5a75 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -20,6 +20,7 @@ package() {
   install -dm755 "${pkgdir}"/usr/lib
   [ "$CARCH" = "i686" ] && install -m644 "${srcdir}"/usr/lib/* "${pkgdir}"/usr/lib
   [ "$CARCH" = "x86_64" ] && install -m644 "${srcdir}"/usr/lib64/* "${pkgdir}"/usr/lib
+  [ "$CARCH" = "loong64" ] && install -m644 "${srcdir}"/usr/lib64/* "${pkgdir}"/usr/lib
   install -Dm644 "${srcdir}"/usr//local/Cg/docs/license.txt "${pkgdir}"/usr/share/licenses/nvidia-cg-toolkit/license.txt
   cp -r "${srcdir}"/usr/local/Cg "${pkgdir}"/usr/share/
   find $pkgdir/usr/share/ -type d -exec chmod -R 755 '{}' ';'
