diff --git a/PKGBUILD b/PKGBUILD
index 346e94e..41817b7 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -25,8 +25,15 @@ optdepends=(
   'xorg-xrandr: Resolution detection (Multi Monitor + Refresh rates)'
   'xorg-xwininfo: See https://github.com/dylanaraps/neofetch/wiki/Images-in-the-terminal'
 )
-source=("${pkgname}-${pkgver}.tar.gz::https://github.com/dylanaraps/neofetch/archive/${pkgver}.tar.gz")
-sha256sums=('58a95e6b714e41efc804eca389a223309169b2def35e57fa934482a6b47c27e7')
+source=("${pkgname}-${pkgver}.tar.gz::https://github.com/dylanaraps/neofetch/archive/${pkgver}.tar.gz"
+        neofetch-la64.patch)
+sha256sums=('58a95e6b714e41efc804eca389a223309169b2def35e57fa934482a6b47c27e7'
+            'ef425de4bd5350b32af8596346201ad9531b72614a4cbbe14128e472131b6ec9')
+
+prepare() {
+  cd "${pkgname}-${pkgver}/"
+  patch -p1 -i "$srcdir/neofetch-la64.patch"
+}
 
 package() {
   cd "${pkgname}-${pkgver}/"
