diff --git a/PKGBUILD b/PKGBUILD
index c79b1be..bdffd5f 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -13,6 +13,7 @@ license=('APACHE' 'LGPL')
 depends=('gcc-libs>=4.7.1-5' 'zlib')
 makedepends=('cmake' 'boost>=1.54.0')
 source=(https://downloads.sourceforge.net/$pkgname/$pkgname-core-$pkgver.tar.gz
+        https://gitweb.gentoo.org/repo/gentoo.git/plain/dev-cpp/clucene/files/clucene-2.3.3.4-gmtime.patch
         # Fedora patches
         clucene-core-2.3.3.4-pkgconfig.patch
         clucene-core-2.3.3.4-install_contribs_lib.patch
@@ -26,6 +27,7 @@ source=(https://downloads.sourceforge.net/$pkgname/$pkgname-core-$pkgver.tar.gz
         clucene-narrowing-conversions.patch
         clucene-multimap-put.patch)
 sha512sums=('1c9da9077edcebd46563bd9e47d330518e0b30061016650a759cfe051e9748fdad8932a472b1cca53a6adafed5f41656527271fc5f55ddfcefb558f0d83286b4'
+            '1c23c08edf5512c29a061e4a8515dfa209151af83e46680842f0aeee1269d529a14fd4a89aab7c25312fd7dbb9daf80912a265d21fcf08ac892a467be4a59a60'
             'b357cb5a1c5d66219f3168724a71af3ebf5c45c752a612d3a69c170b739acc065dc17c261c3a730298ea6c637fe820637a100f73ab03d931734f80bb598fbf55'
             '0aa92635949089196e4e7c579e78761e8751987ef5036320a161a4aaa67da3c63756398c903419c76ea0fbdc8a949e871fcb65be98179a9853a24a5a4cacfde3'
             'f606481b3bae44487a05e81da1e19dfa0bc5db8b10832d5b84c4e269fecb99ad010b90c5132e618c300f32b8c5bf28cfd0038c4ca2ddb4870c5a3f5113a18e64'
@@ -51,6 +53,7 @@ prepare() {
   patch -Np0 -i "${srcdir}"/clucene-debug.patch
   patch -Np0 -i "${srcdir}"/clucene-narrowing-conversions.patch
   patch -Np0 -i "${srcdir}"/clucene-multimap-put.patch
+  patch -Np1 -i "${srcdir}"/clucene-2.3.3.4-gmtime.patch
 
   # FS#77036 / https://sourceforge.net/p/clucene/bugs/235/ 
   patch -Np1 -i ../0001-Fix-missing-include-time.h.patch
