diff --git a/PKGBUILD b/PKGBUILD
index 2ec7df6..f56790d 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -81,7 +81,7 @@ build() {
  ./configure --prefix=/usr \
              --enable-qt5 \
              --disable-qt4 \
-             --enable-hpcups-install \
+             --disable-hpcups-install \
              --enable-cups-drv-install \
              --disable-imageProcessor-build \
              --enable-pp-build #--help
