diff --git a/PKGBUILD b/PKGBUILD
index 14cc184..c929515 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -4,22 +4,26 @@
 # Contributor: Keshav Amburay <(the ddoott ridikulus ddoott rat) (aatt) (gemmaeiil) (ddoott) (ccoomm)>
 
 ## "1" to enable IA32-EFI build in Arch x86_64, "0" to disable
-_IA32_EFI_IN_ARCH_X64="1"
+_IA32_EFI_IN_ARCH_X64="0"
 
 ## "1" to enable EMU build, "0" to disable
 _GRUB_EMU_BUILD="0"
+SKIPCONFIG=1
 
 [[ "${CARCH}" == 'x86_64' ]] && _EFI_ARCH='x86_64'
 [[ "${CARCH}" == 'i686' ]] && _EFI_ARCH='i386'
+[[ "${CARCH}" == 'loong64' ]] && _EFI_ARCH='loongarch64'
 
 [[ "${CARCH}" == 'x86_64' ]] && _EMU_ARCH='x86_64'
 [[ "${CARCH}" == 'i686' ]] && _EMU_ARCH='i386'
+[[ "${CARCH}" == 'loong64' ]] && _EMU_ARCH=''
 
 pkgname='grub'
 pkgdesc='GNU GRand Unified Bootloader (2)'
 epoch=2
+_tag='03e6ea18f6f834f177cad017279bedbb0a3de594' # git rev-parse grub-${_pkgver}
 _pkgver=2.12
-_unifont_ver='15.1.05'
+_unifont_ver='15.1.04'
 pkgver=${_pkgver/-/}
 pkgrel=2
 url='https://www.gnu.org/software/grub/'
@@ -56,7 +60,7 @@ validpgpkeys=('E53D497F3FA42AD8C9B4D1E835A93B74E82E4209'  # Vladimir 'phcoder' S
               'BE5C23209ACDDACEB20DB0A28C8189F1988C2166'  # Daniel Kiper <dkiper@net-space.pl>
               '95D2E9AB8740D8046387FD151A09227B1F435A33') # Paul Hardy <unifoundry@unifoundry.com>
 
-source=("git+https://git.savannah.gnu.org/git/grub.git#tag=grub-${_pkgver}?signed"
+source=("git+https://git.savannah.gnu.org/git/grub.git#tag=${_tag}?signed"
         'git+https://git.savannah.gnu.org/git/gnulib.git'
         "https://ftp.gnu.org/gnu/unifont/unifont-${_unifont_ver}/unifont-${_unifont_ver}.bdf.gz"{,.sig}
         '0001-00_header-add-GRUB_COLOR_-variables.patch'
@@ -67,7 +71,7 @@ source=("git+https://git.savannah.gnu.org/git/grub.git#tag=grub-${_pkgver}?signe
 
 sha256sums=('SKIP'
             'SKIP'
-            '8ea5b5a14d71e3353d1fea373f5d88d198ad1e285cedd8294655926ee11fd91d'
+            '88e00954b10528407e62e97ce6eaa88c847ebfd9a464cafde6bf55c7e4eeed54'
             'SKIP'
             '5dee6628c48eef79812bb9e86ee772068d85e7fcebbd2b2b8d1e19d24eda9dab'
             '8488aec30a93e8fe66c23ef8c23aefda39c38389530e9e73ba3fbcc8315d244d'
@@ -244,8 +248,10 @@ _build_grub-emu() {
 build() {
 	cd "${srcdir}/grub/"
 
-	echo "Build grub bios stuff..."
-	_build_grub-common_and_bios
+	if [[ "${CARCH}" != "loong64" ]]; then
+	    echo "Build grub bios stuff..."
+	    _build_grub-common_and_bios
+	fi
 
 	echo "Build grub ${_EFI_ARCH} efi stuff..."
 	_build_grub-efi
@@ -318,6 +324,15 @@ package() {
 		_package_grub-emu
 	fi
 
-	echo "Package grub bios stuff..."
-	_package_grub-common_and_bios
+	if [[ "${CARCH}" != "loong64" ]]; then
+	    echo "Package grub bios stuff..."
+	    _package_grub-common_and_bios
+	    cd "${srcdir}/grub-bios/"
+
+	    echo "Run make install for bios build..."
+	    make DESTDIR="${pkgdir}/" bashcompletiondir="/usr/share/bash-completion/completions" install
+	fi
+
+	echo "Install /etc/default/grub (used by grub-mkconfig)..."
+	install -D -m0644 "${srcdir}/grub.default" "${pkgdir}/etc/default/grub"
 }
