diff --git a/PKGBUILD b/PKGBUILD
index 04bebc4..dcaaa2d 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -64,7 +64,7 @@ package() {
   done
 
   # Move into the target directory
-  cd "target/products/org.jkiss.dbeaver.core.product/linux/gtk/${CARCH}"
+  cd "target/products/org.jkiss.dbeaver.core.product/linux/gtk/`uname -m`"
 
   # Initially install everything into /usr/lib/dbeaver
   install -m 755 -d "${pkgdir}/usr/lib"
