diff --git a/PKGBUILD b/PKGBUILD
index c4980e2..b22484c 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -47,11 +47,11 @@ _commit='1941c1d5abf81814992b68bbc86c0020dc75a3ad'
 source=("$pkgname::git+https://github.com/pypa/virtualenv#commit=$_commit")
 b2sums=('SKIP')
 
-pkgver() {
-  cd "$pkgname"
-
-  git describe --tags | sed 's/^v//'
-}
+#pkgver() {
+#  cd "$pkgname"
+#
+#  git describe --tags | sed 's/^v//'
+#}
 
 build() {
   cd "$pkgname"
