diff --git a/PKGBUILD b/PKGBUILD
index cfe8a60..ddf0e68 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -15,14 +15,14 @@ sha256sums=('e5839d74071c44efb44ae33859ff438ff5823c007960889f567b2c2c33cff4d1')
 
 prepare() {
   cd "$pkgname-$pkgver"
-  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+  cargo fetch --target "`uname -m`-unknown-linux-gnu"
 }
 
 build() {
   cd "$pkgname-$pkgver"
   export LIBSSH2_SYS_USE_PKG_CONFIG=1
   CFLAGS+=" -ffat-lto-objects"
-  cargo build --release --frozen
+  cargo build --release
 }
 
 # Tests require rustup nightly
