diff --git a/PKGBUILD b/PKGBUILD
index a1b5a16..f43c8c5 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -25,6 +25,8 @@ options=('!makeflags')
 
 prepare() {
 	cd "${srcdir}/${pkgname}${_pkgver}"
+    CFLAGS=${CFLAGS/-Wformat -Werror=format-security/}
+    CXXFLAGS=${CXXFLAGS/-Wformat -Werror=format-security/}
 	sed -e "/^CFLAGS_NOOPT =/s/\$/ $CPPFLAGS $CFLAGS/" -i unix/Makefile
 	sed -e "s/^LFLAGS1=''/LFLAGS1=$LDFLAGS/" -i unix/configure
 	patch -p1 -i ../zip-3.0-currdir.patch
