diff --git a/PKGBUILD b/PKGBUILD
index 2d8eb4f..9aac6c9 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -42,8 +42,10 @@ groups=(gnome)
 _commit=c3f8ef3d7b4ec57b948992ddd8137db02f9d25b7  # tags/3.6.2^0
 source=(
   "git+https://gitlab.gnome.org/GNOME/tracker-miners.git#commit=$_commit"
+  tracker-miners-fix-build.patch
 )
-b2sums=('SKIP')
+b2sums=('SKIP'
+        'f9896f048312e8e5dd080d3fa07330abbf33277ef596db55e0693948dc823d3526ecc700307c35324ac0d95c1710af90a7237299baaef32e0d87858ebb7521ef')
 
 pkgver() {
   cd tracker-miners
