diff --git a/PKGBUILD b/PKGBUILD
index 18340d6..0c3f611 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -8,43 +8,37 @@ pkgver=1.0.9
 pkgrel=9
 pkgdesc='GTK/Gnome interface around MPlayer'
 arch=('x86_64')
-url='https://github.com/kdekorte/gnome-mplayer'
-license=('GPL-2.0-or-later')
-depends=('alsa-lib' 'cairo' 'dconf' 'dbus' 'dbus-glib' 'gdk-pixbuf2' 'glib2' 'glibc' 'gmtk' 'gtk3'
-         'hicolor-icon-theme' 'libnotify' 'libx11' 'libxss')
-makedepends=('git')
-source=("git+https://github.com/kdekorte/gnome-mplayer.git#tag=v$pkgver"
-        'gnome-mplayer.appdata.xml')
-b2sums=('eb4a2008ba86f9ba99aa5b96e590212c7d3d15594f618276da08e2c1434962a0d62ff5215ea7e88214565ca22b96fa3bbd33791be51044b000970871a4fccf8f'
-        '63427949e71ce98e450cc854be1d79e8076eafc4a2d854a3b5edaf25482f37a8cf189782db84ff81ad8a32023d21b148913ac7b8e2f3d9a7de4ba53741c64909')
+url='https://sites.google.com/site/kdekorte2/gnomemplayer'
+license=('GPL')
+depends=('dbus-glib' 'libgda' 'libgpod' 'libnotify' 'gmtk')
+makedepends=('caja' 'nautilus' 'nemo' 'python')
+optdepends=('caja: file manager integration for Caja'
+            'nautilus: file manager integration for Nautilus'
+            'nemo: file manager integration for Nemo')
+source=("$pkgname-$pkgver.tar.gz::https://github.com/kdekorte/$pkgname/archive/v$pkgver.tar.gz"
+        "gnome-mplayer.appdata.xml"
+        "caja.patch")
+sha256sums=('07e11bc42834dae14ad66f887e6045f2bea94eae7f797b624c9a099f806b5cde'
+            '2e661347fa2765a9553938db9582a833b42385db92f6687172507deecf404567'
+            '52dbf11e64042c03ec94677ee667d88306f5f45521b7a1da82e5e5f02f59ed28')
 
 prepare() {
-  cd $pkgname
-
-  # Add support for mate screensaver
-  git cherry-pick -n 626caf4cf2ea9c69ea27113d0087dc3871d10169
-
-  # Fix incompatible pointer type
-  sed -i 's/gtk_check_menu_item_get_active(menuitem_view_subtitles)/gtk_check_menu_item_get_active(GTK_CHECK_MENU_ITEM(menuitem_view_subtitles))/' src/gui.c
-
-  # Remove leftover of gconf
+  cd $pkgname-$pkgver
+  patch -Np0 -i ../caja.patch
+  sed -i 's/Exec=gnome-mplayer %U/Exec=gnome-mplayer %F/' $pkgname.desktop
   sed -i '/AM_GCONF_SOURCE_2/d' configure.in
-
-  # Disable Wayland backend as it's not supported
-  sed -i '/g_type_init();/a gdk_set_allowed_backends ("x11");' src/main.c
-
   autoreconf -fi
 }
 
 build() {
-  cd $pkgname
+  cd $pkgname-$pkgver
   CFLAGS+=' -fcommon' # https://wiki.gentoo.org/wiki/Gcc_10_porting_notes/fno_common
   ./configure --prefix=/usr
   make
 }
 
 package() {
-  cd $pkgname
+  cd $pkgname-$pkgver
   make DESTDIR="$pkgdir" install
   install -Dm644 ../gnome-mplayer.appdata.xml "$pkgdir/usr/share/metainfo/gnome-mplayer.appdata.xml"
 }
diff --git a/gnome-mplayer.appdata.xml b/gnome-mplayer.appdata.xml
index e139b4b..216e092 100644
--- a/gnome-mplayer.appdata.xml
+++ b/gnome-mplayer.appdata.xml
@@ -5,7 +5,7 @@
   <name>GNOME MPlayer</name>
   <summary>Play your media</summary>
   <metadata_license>CC0-1.0</metadata_license>
-  <project_license>GPL-2.0-or-later</project_license>
+  <project_license>GPL-2.0</project_license>
   <developer_name>Kevin DeKorte</developer_name>
   <description>
     <p>A GTK2/GTK3 interface to MPlayer. The power of MPlayer combined with a friendly interface for your desktop; You can play all your multimedia (audio, video, CD, DVDs, and VCDs, streams etc.), organize, sort and create playlists, take screenshots while playing videos, be notified about media changes. Full DVD and MKV chapter support, when supported by Mplayer. Subtitle support with the ability to specify preferred audio and subtitle languages if the media supports it. Support for cover art retrieval from Amazon.com for audio media files with artist and/or album information contained in the file.</p>
@@ -16,7 +16,7 @@
     </screenshot>
   </screenshots>
   <url type="bugtracker">https://github.com/kdekorte/gnome-mplayer/issues</url>
-  <url type="homepage">https://github.com/kdekorte/gnome-mplayer</url>
+  <url type="homepage">https://sites.google.com/site/kdekorte2/gnomemplayer</url>
   <update_contact>gnome-mplayer@googlegroups.com</update_contact>
   <translation type="gettext">gnome-mplayer</translation>
 </component>
