diff --git a/PKGBUILD b/PKGBUILD
index dd6721b..cb5a224 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -18,7 +18,7 @@ sha512sums=('55779ad0c01e065678e1a57f338272f1d38057658fe6b7c54f7bc35595575aafe13
 prepare() {
   cd "$pkgname-$pkgver"
   patch -Np1 -i "../$pkgname-bash-completion.patch"
-  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+  cargo fetch --locked --target "`uname -m`-unknown-linux-gnu"
 }
 
 build() {
