diff --git a/PKGBUILD b/PKGBUILD
index 9652128..f870109 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -22,11 +22,17 @@ sha256sums=('19cbeb9964ebe439413dd66d82ace1f904adc5f25d8a823c1b48c34bd0d29ea5')
 
 build() {
   cd SuiteSparse-$pkgver
+  CXXFLAGS=${CXXFLAGS/-mlsx /}
+  CXXFLAGS=${CXXFLAGS/-fstack-clash-protection/}
+
+  CFLAGS+=" -ffat-lto-objects -mcmodel=extreme" \
+  CXXFLAGS+=" -ffat-lto-objects -mcmodel=extreme" \
   CMAKE_OPTIONS="-DBLA_VENDOR=Generic \
                  -DCMAKE_INSTALL_PREFIX=/usr \
                  -DCMAKE_BUILD_TYPE=None \
                  -DNSTATIC=ON" \
   make
+#  -DALLOW_64BIT_BLAS=ON \
 }
 
 package() {
