diff --git a/PKGBUILD b/PKGBUILD
index 0d28799..6ffcb90 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -23,6 +23,12 @@ export CARGO_PROFILE_RELEASE_DEBUG=2
 
 build() {
   arch-meson $pkgname-$pkgver build
+  find -name Cargo.lock -exec rm -f {} \;
+  mkdir -p .cargo
+  cat > .cargo/config.toml <<EOF
+[source.crates-io]
+registry = "https://gitee.com/yetist/crates.io-index"
+EOF
   meson compile -C build
 }
 
