diff --git a/0001-HACK-Don-t-run-fuzzing-tests.patch b/0001-HACK-Don-t-run-fuzzing-tests.patch
index 0ead9a1..0bce92a 100644
--- a/0001-HACK-Don-t-run-fuzzing-tests.patch
+++ b/0001-HACK-Don-t-run-fuzzing-tests.patch
@@ -9,7 +9,7 @@ They take way too long.
  1 file changed, 1 insertion(+), 4 deletions(-)
 
 diff --git a/Makefile.am b/Makefile.am
-index 0a49d37547b8..f1bc8deb3364 100644
+index 5bc401813e6b..0d69d2cdf235 100644
 --- a/Makefile.am
 +++ b/Makefile.am
 @@ -6,11 +6,8 @@ SUBDIRS = include . doc example xstc
diff --git a/PKGBUILD b/PKGBUILD
index adb395f..e5b2598 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -33,7 +33,7 @@ source=(
   https://www.w3.org/XML/Test/xmlts20130923.tar.gz
 )
 b2sums=('SKIP'
-        '6931b07a467874c3016deb7a68bab1e9cfaabc88fdeaf0ccf6c9cca02bbce9618b18f8c285f5748089138489d04d7c72a659b2bfd4ea5fe24ad273daacbd6d50'
+        '5f47a4949e9485c40b5685840d48aef87aa7919eb7e0afd679b598ee054392f18a657947c10c627d3239937e53611f8263c47312d26b093aa5973854307f53aa'
         '63a47bc69278ef510cd0b3779aed729e1b309e30efa0015d28ed051cc03f9dfddb447ab57b07b3393e8f47393d15473b0e199c34cb1f5f746b15ddfaa55670be')
 
 pkgver() {
@@ -58,15 +58,15 @@ build() {
     --prefix=/usr
     --sysconfdir=/etc
     --localstatedir=/var
-    --disable-static
     --with-history
     --with-icu
-    --with-legacy
     --with-python=/usr/bin/python
     --with-threads
+    --disable-static
   )
 
   cd libxml2
+
   ./configure "${configure_options[@]}"
   sed -i -e 's/ -shared / -Wl,-O1,--as-needed\0/g' libtool
   make
@@ -82,6 +82,7 @@ package_libxml2() {
   provides=(libxml2.so)
 
   cd libxml2
+
   make DESTDIR="$pkgdir" install
 
   mkdir -p ../doc/usr/share
