diff --git a/PKGBUILD b/PKGBUILD
index 85cb532..61d1a22 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -83,7 +83,7 @@ package() {
   cd "$pkgname"
 
   # plugin
-  install -vDm755 -t "$pkgdir/usr/lib/$pkgname" "build/reaper_reapack-$CARCH.so"
+  install -vDm755 -t "$pkgdir/usr/lib/$pkgname" "build/reaper_reapack-`uname -m`.so"
 
   # documentation
   install -vDm644 -t "$pkgdir/usr/share/doc/$pkgname" README.md
