diff --git a/PKGBUILD b/PKGBUILD
index d4e0b19..9103fab 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -36,6 +36,7 @@ sha256sums=('SKIP'
             '6a4ec2bfcf4e7cff73346762b252cc28be5ca0ed79fde5e69350efe229b43adc'
             '5411edbe215c24b30448fac69bd0ba7c882f545e8cf05027b2b6e2227abc5e78'
             '4abc22150fa3e06b2fdcec32146abc9be4e316692aa4d5bd5aa53b4b726783fa')
+options=(debug)
 
 pkgver() {
   cd $_pkgfqn
