diff --git a/PKGBUILD b/PKGBUILD
index 4903c25..2b3dfcf 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -14,6 +14,9 @@ sha512sums=('SKIP')
 
 build() {
   cd domain-list-community
+  go mod edit -replace=golang.org/x/sys=github.com/golang/sys@v0.0.0-20220622161953-175b2fd9d664
+  go mod edit -replace=golang.org/x/net=github.com/golang/net@v0.0.0-20220622184535-263ec571b305
+  go mod tidy
   ASSUME_NO_MOVING_GC_UNSAFE_RISK_IT_WITH=go1.18 go run main.go
 }
 
