diff --git a/PKGBUILD b/PKGBUILD
index ae460cb..e84889e 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -37,7 +37,7 @@ package() {
   rm -vrf cache
   cd "gems/$_gemname-$pkgver"
   rm -vrf Gemfile Rakefile "$_gemname.gemspec" ext rakelib
-  find lib/ffi/platform/* -prune -not -name "$CARCH-linux" -exec rm -rf {} +
+  find lib/ffi/platform/* -prune -not -name "`uname -m`-linux" -exec rm -rf {} +
 
   # move documentation
   install -vd "$pkgdir/usr/share/doc/$pkgname"
