diff --git a/PKGBUILD b/PKGBUILD
index c2f04a6..4798df8 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -12,8 +12,8 @@ depends=('glibc' 'imagemagick' 'ghostscript')
 makedepends=('texinfo' 'texlive-bin' 'texlive-plainextra')
 checkdepends=('texlive-latexextra') # for a4wide package
 source=("https://downloads.sourceforge.net/sourceforge/${pkgname}/${pkgname}-${pkgver}a.tar.gz")
-md5sums=('b984073a9481eae26760ce03015d0373')
-sha1sums=('7fe41c82465921405bbbc6667694bf2731c7dfd1')
+md5sums=('404428600d0d4c0dbdec5b6981727050')
+sha1sums=('4636fe6f27516f759d67944fe42be646c97ba27a')
 
 build() {
   cd "${srcdir}/${pkgname}-${pkgver}"
