diff --git a/PKGBUILD b/PKGBUILD
index e26f85d..cd8bf01 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -10,7 +10,7 @@ pkgdesc='Library to retrieve information on the current status of computer hardw
 arch=(x86_64)
 url='https://kde.org/plasma-desktop/'
 license=(LGPL)
-depends=(libxres qt5-webengine knewstuff5)
+depends=(libxres knewstuff5)
 makedepends=(extra-cmake-modules kdoctools5 qt5-tools)
 conflicts=('ksysguard<5.21.90')
 groups=(plasma)
