diff --git a/PKGBUILD b/PKGBUILD
index d9c9a13..a68f59f 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -9,10 +9,10 @@ pkgdesc='Meta package providing the latest available stable Electron build'
 arch=(any)
 url='https://electronjs.org'
 license=(MIT)
-depends=("electron$pkgver")
+depends=("electron$pkgver-bin")
 
 package() {
 	mkdir -p "$pkgdir/usr/bin" "$pkgdir/usr/lib"
-	ln -sf "${depends[0]}" "$pkgdir/usr/bin/$pkgname"
-	ln -sf "${depends[0]}" "$pkgdir/usr/lib/$pkgname"
+	ln -sf "${depends[0]%-bin}" "$pkgdir/usr/bin/$pkgname"
+	ln -sf "${depends[0]%-bin}" "$pkgdir/usr/lib/$pkgname"
 }
