diff --git a/PKGBUILD b/PKGBUILD
index aae2119..f916668 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -16,6 +16,7 @@ source=("https://github.com/cisco/openh264/archive/v${pkgver}/${pkgname}-${pkgve
         "https://github.com/google/googletest/archive/release-${_gtestver}/googletest-${_gtestver}.tar.gz")
 sha256sums=('a44d1ccc348a790f9a272bba2d1c2eb9a9bbd0302e4e9b655d709e1c32f92691'
             '9bf1fe5182a604b4135edc1a425ae356c9ad15e9b23f9f12a02e80184c3a249c')
+options=(!lto)
 
 prepare() {
     ln -sf "../googletest-release-${_gtestver}" "${pkgname}-${pkgver}/gtest"
