diff --git a/PKGBUILD b/PKGBUILD
index bd546d1..bb9b6ab 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -6,10 +6,11 @@ pkgver=3.0.1
 pkgrel=8
 pkgdesc="PDF printer for cups"
 arch=(x86_64)
-depends=('cups' 'libcups' 'ghostscript' 'glibc')
+depends=('cups' 'ghostscript')
 install=cups-pdf.install
 url="https://www.cups-pdf.de/welcome.shtml"
-license=('GPL-2.0-or-later')
+license=('GPL2')
+options=('debug')
 source=(https://www.cups-pdf.de/src/cups-pdf_$pkgver.tar.gz
         remove-deprecated-ghostscript-setpdfwrite-operator.diff)
 backup=(etc/cups/cups-pdf.conf)
