diff --git a/PKGBUILD b/PKGBUILD
index 0cd43f0..6071f45 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -28,7 +28,7 @@ check() {
   gcc ${CFLAGS} ${CPPFLAGS} ${LDFLAGS} -fPIC -shared -o testextension.sqlext src/testextension.c
   # do glob expansion in variable assignment
   local python_version=$(python -c 'import sys; print("".join(map(str, sys.version_info[:2])))')
-  PYTHONPATH="$PWD/build/lib.linux-$CARCH-cpython-${python_version}" python -m unittest discover -vs .
+  PYTHONPATH="$PWD/build/lib.linux-`uname -m`-cpython-${python_version}" python -m unittest discover -vs .
 }
 
 package() {
