diff --git a/PKGBUILD b/PKGBUILD
index 9bac50b..a9e7cf9 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -37,7 +37,7 @@ package() {
   python setup.py install --root="$pkgdir" --optimize=1 --skip-build
   install -Dm 644 LICENSE.txt -t "$pkgdir/usr/share/licenses/$pkgname"
   install -Dm 644 CHANGELOG.rst README.rst docs/_build/text/*.txt -t "$pkgdir/usr/share/doc/$pkgname"
-  install -Dm 644 docs/_build/man/parso.1 "$pkgdir/usr/share/man/man1/$pkgname.1"
+  #install -Dm 644 docs/_build/man/parso.1 "$pkgdir/usr/share/man/man1/$pkgname.1"
 }
 
 # vim: ts=2 sw=2 et:
