diff --git a/PKGBUILD b/PKGBUILD
index 705d271..b2b7adc 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -4,7 +4,7 @@ pkgname=bucklespring
 pkgver=1.5.1
 pkgrel=1
 pkgdesc="Nostalgia bucklespring keyboard sound, sampled from IBM's Model-M"
-arch=('i686' 'x86_64')
+arch=('i686' 'loong64' 'x86_64')
 url="https://github.com/zevv/bucklespring"
 license=('MIT')
 depends=('openal' 'alure' 'libxtst' 'dumb' 'mpg123')
