diff --git a/PKGBUILD b/PKGBUILD
index 8b1f32c..98eb185 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -46,11 +46,11 @@ source=(
 b2sums=('SKIP'
         '04dad8ce369f4fb78e2b3239e30d63410d8e8c33b5e05717b5b62ba54c624c2e931220b950bad507deab60a07086c434949f0766bc3ae2eb771d62e1919cbecd')
 
-pkgver() {
-  cd "$pkgname"
-
-  git describe --tags | sed 's/^rel\.//'
-}
+#pkgver() {
+#  cd "$pkgname"
+#
+#  git describe --tags | sed 's/^rel\.//' |sed 's/-/./g'
+#}
 
 prepare() {
   cd "$pkgname"
