diff --git a/PKGBUILD b/PKGBUILD
index 1f0e8d7..35f99db 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -19,7 +19,7 @@ b2sums=('d8571ddf5ab3684206e4802871aedea216f6edc978adcb80b7879e58aa01d564634ec65
 prepare() {
   cd "embuild-${pkgname}-v${pkgver}/${pkgname}"
   cp ../../Cargo.lock ..
-  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+  cargo fetch --locked --target "`uname -m`-unknown-linux-gnu"
 }
 
 updlockfiles() {
