diff --git a/PKGBUILD b/PKGBUILD
index c295a08..fe7f6ef 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -21,7 +21,13 @@ prepare() {
 
 build() {
   cd "$srcdir/$pkgname-$pkgver"
-  cargo build --release --locked
+  find -name Cargo.lock -exec rm -f {} \;
+  mkdir -p .cargo
+  cat > .cargo/config.toml <<EOF
+[source.crates-io]
+registry = "https://gitee.com/yetist/crates.io-index"
+EOF
+  cargo build --release
 }
 
 check() {
