diff --git a/PKGBUILD b/PKGBUILD
index 1923663..de86f71 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -19,6 +19,8 @@ sha256sums=('da68d7a65f44dcf6ce6e4e630b6f6dd9897249d34425920bfdd4e07ff1866a72'
 build() {
 	cd "${srcdir}/${pkgname}-${pkgver}"
 	CFLAGS+=' -fcommon' # https://wiki.gentoo.org/wiki/Gcc_10_porting_notes/fno_common
+    CFLAGS=${CFLAGS/-Werror=format-security/}
+    CXXFLAGS=${CXXFLAGS/-Werror=format-security/}
 	./configure --prefix=/usr
 	make
 }
