diff --git a/PKGBUILD b/PKGBUILD
index c8b6617..c3d0829 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -109,7 +109,7 @@ build() {
 check() {
   cd "$srcdir/$pkgbase-$pkgver"
   local _pyver=$(python -c "import sys; print('{0}{1}'.format(*sys.version_info[:2]))")
-  PYTHONPATH="pyb/lib.linux-$CARCH-cpython-$_pyver" python -c 'import grpc'
+  PYTHONPATH="pyb/lib.linux-`uname -m`-cpython-$_pyver" python -c 'import grpc'
 }
 
 package_grpc() {
