diff --git a/PKGBUILD b/PKGBUILD
index 8cc97e4..457f1a5 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -35,6 +35,8 @@ prepare() {
 
 build() {
   cd $pkgname
+  CFLAGS=${CFLAGS/-Werror=format-security/}
+  CXXFLAGS=${CXXFLAGS/-Werror=format-security/}
   ./configure --prefix=/usr --sysconfdir=/etc --localstatedir=/var \
     --disable-vcd-info --disable-cddb --enable-cpp-progs --disable-static \
     --enable-maintainer-mode
