diff --git a/PKGBUILD b/PKGBUILD
index b4ca32a..3e85089 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -17,7 +17,7 @@ makedepends=(lua
              lua53
              luarocks)
 _archive="$pkgbase-$_pkgver"
-_rock="$pkgbase-$pkgver-$_rockrel.linux-$CARCH.rock"
+_rock="$pkgbase-$pkgver-$_rockrel.linux-`uname -m`.rock"
 _rockspec="$pkgbase-$pkgver-$_rockrel.rockspec"
 source=("$url/archive/$_pkgver/$_archive.tar.gz")
 sha256sums=('8ff94f390ea9d98c734699373ca3b0ce500d651b2ab1cb8d7d2336fc5b79cded')
