diff --git a/PKGBUILD b/PKGBUILD
index 86e4feb..2b261bc 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -58,8 +58,8 @@ _nextcloud_app_package() {
 
 prepare() {
 	cd "$_archive"
-	cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
-	sed -i -e "s/@ARCH@/$CARCH/" "../$pkgname.service"
+	cargo fetch --locked --target "`uname -m`-unknown-linux-gnu"
+	sed -i -e "s/@ARCH@/`uname -m`/" "../$pkgname.service"
 }
 
 build() {
@@ -89,6 +89,6 @@ package() {
 		--include=webpack.prod.js \
 		--exclude='*' \
 		./ $_appdir/
-	install -Dm0755 -t "$_appdir/bin/$CARCH/" "target/release/${pkgname##*-}"
+	install -Dm0755 -t "$_appdir/bin/`uname -m`/" "target/release/${pkgname##*-}"
 	install -Dm0644 -t "$pkgdir/usr/lib/systemd/system/" "$srcdir/$pkgname.service"
 }
