diff --git a/PKGBUILD b/PKGBUILD
index c680b40..45479b9 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -4,7 +4,7 @@ pkgname=mingw-w64-gcc
 pkgver=13.1.0
 pkgrel=1
 pkgdesc="Cross GCC for the MinGW-w64 cross-compiler"
-arch=('x86_64' 'aarch64')
+arch=('loong64' 'x86_64' 'aarch64')
 url="https://gcc.gnu.org"
 license=('GPL' 'LGPL' 'FDL' 'custom')
 groups=('mingw-w64-toolchain' 'mingw-w64')
@@ -34,7 +34,7 @@ build() {
     "$srcdir"/gcc/configure --prefix=/usr --libexecdir=/usr/lib \
         --target=${_arch} \
         --with-bugurl=https://bugs.archlinux.org/ \
-        --enable-languages=ada,c,c++,fortran,lto,objc,obj-c++ \
+        --enable-languages=c,c++,fortran,lto,objc,obj-c++ \
         --enable-shared --enable-static \
         --enable-threads=posix --enable-fully-dynamic-string \
         --enable-libstdcxx-time=yes --enable-libstdcxx-filesystem-ts=yes \
@@ -52,7 +52,7 @@ package() {
     make DESTDIR="$pkgdir" install
     ${_arch}-strip "$pkgdir"/usr/${_arch}/lib/*.dll
     strip "$pkgdir"/usr/bin/${_arch}-*
-    strip "$pkgdir"/usr/lib/gcc/${_arch}/${pkgver}/{cc1*,collect2,gnat1,f951,lto*}
+    strip "$pkgdir"/usr/lib/gcc/${_arch}/${pkgver}/{cc1*,collect2,f951,lto*}
     ln -s ${_arch}-gcc "$pkgdir"/usr/bin/${_arch}-cc
     # mv dlls
     mkdir -p "$pkgdir"/usr/${_arch}/bin/
