diff --git a/PKGBUILD b/PKGBUILD
index 4b00932..44ca70e 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -31,6 +31,9 @@ check() {
 
 build() {
   cd memcached_exporter-$pkgver
+  go mod edit -replace=golang.org/x/sys=github.com/golang/sys@v0.0.0-20220622161953-175b2fd9d664
+  go mod edit -replace=golang.org/x/net=github.com/golang/net@v0.0.0-20220622184535-263ec571b305
+  go mod tidy
   go build \
     -trimpath \
     -buildmode=pie \
