diff --git a/PKGBUILD b/PKGBUILD
index 593d989..7c04807 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -24,14 +24,23 @@ makedepends=(git go)
 replaces=(go-pie)
 provides=(go-pie)
 options=(!strip staticlibs)
-source=(https://go.dev/dl/go${pkgver}.src.tar.gz{,.asc})
+source=(https://go.dev/dl/go${pkgver}.src.tar.gz{,.asc}
+http://public.loongarch.dev/sources/go/go1.21p.tar.gz/9a6b12102fddd56e29e41a67d0494166/go1.21p.tar.gz)
 validpgpkeys=('EB4C1BFD4F042F6DDDCCEC917721F63BD38B4796')
 sha256sums=('124926a62e45f78daabbaedb9c011d97633186a33c238ffc1e25320c02046248'
-            'SKIP')
+            'SKIP'
+            'a0ec920455ec49777d4bba8ce64f80b6c93458e62112c7178fd7ac34ecbfb506')
+
+prepare() {
+  cd "$pkgname"
+  for i in $srcdir/go1.21p/*.patch;
+  do
+    patch -p1 -i $i
+  done
+}
 
 build() {
-  export GOARCH=amd64
-  export GOAMD64=v1 # make sure we're building for the right x86-64 version
+  export GOARCH=loong64
   export GOROOT_FINAL=/usr/lib/go
   export GOROOT_BOOTSTRAP=/usr/lib/go
 
@@ -50,7 +59,7 @@ package() {
   cd "$pkgname"
 
   install -d "$pkgdir/usr/bin" "$pkgdir/usr/lib/go" "$pkgdir/usr/share/doc/go" \
-    "$pkgdir/usr/lib/go/pkg/linux_amd64_"{dynlink,race}
+    "$pkgdir/usr/lib/go/pkg/linux_loong64_"{dynlink,race}
 
   cp -a bin pkg src lib misc api test "$pkgdir/usr/lib/go"
   # We can't strip all binaries and libraries,
