diff --git a/PKGBUILD b/PKGBUILD
index 0e21027..16e6835 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -36,7 +36,7 @@ validpgpkeys=(
 prepare() {
   cd $pkgname
   export RUSTUP_TOOLCHAIN=stable
-  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+#  cargo fetch --locked --target "`uname -m`-unknown-linux-gnu"
 }
 
 build() {
@@ -44,7 +44,7 @@ build() {
   export CARGO_TARGET_DIR=../target
   export RUSTUP_TOOLCHAIN=stable
   # NOTE: we select specific (default) features, as there are multiple crypto backends
-  cargo build --release --frozen --features 'default'
+  cargo build --release --features 'default'
   # create the man pages
   SQ_MAN=$CARGO_TARGET_DIR/manpages cargo run
 
