diff --git a/PKGBUILD b/PKGBUILD
index 4175e04..fc7b25a 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -11,7 +11,6 @@ arch=(x86_64)
 url='https://www.oracle.com/technology/software/products/berkeley-db/index.html'
 license=(AGPL3)
 depends=(gcc-libs sh)
-provides=(libdb.so)
 source=(https://download.oracle.com/berkeley-db/db-${pkgver}.tar.gz)
 b2sums=('d3254fba1300d6c7dda5d872c5fa349e8704d71be9c4d37ccf3357782adeb879ce30e066cf083bd97554405c2e94071a96cee16090ad9f797a4fa657ff7e8c46')
 
