diff --git a/PKGBUILD b/PKGBUILD
index 24d7115..dbd49c8 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -38,7 +38,7 @@ check() {
   # == An ugly hack to generate .ghc.environment without cabal-install ==
   # doctest-parallel relies on this cabal-install feature to configure GHCi.
   # https://github.com/martijnbastiaan/doctest-parallel/issues/22
-  _ghc_env_filename=.ghc.environment.$CARCH-linux-$(expac %v ghc | cut -d - -f 1)
+  _ghc_env_filename=.ghc.environment.`uname -m`-linux-$(expac %v ghc | cut -d - -f 1)
   echo -e "package-db dist/package.conf.inplace" > $_ghc_env_filename
   ls dist/package.conf.inplace/*.conf | sed 's|.*/\(.*\).conf$|package-id \1|' >> $_ghc_env_filename
   ls /usr/lib/ghc-9.0.2/package.conf.d/*.conf | sed 's|.*/\(.*\).conf$|package-id \1|' >> $_ghc_env_filename
