diff --git a/PKGBUILD b/PKGBUILD
index 198da1b..c3ee958 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -8,7 +8,7 @@ url='https://gitlab.archlinux.org/archlinux/arch-rebuild-order'
 arch=('x86_64')
 license=('MIT')
 depends=('glibc' 'libalpm.so')
-makedepends=('cargo' 'mandown' 'git')
+makedepends=('rust' 'mandown' 'git')
 groups=('archlinux-tools')
 source=(git+https://gitlab.archlinux.org/archlinux/arch-rebuild-order.git#tag=v$pkgver?signed)
 sha512sums=('SKIP')
@@ -16,12 +16,12 @@ validpgpkeys=("E499C79F53C96A54E572FEE1C06086337C50773E")
 
 prepare() {
   cd ${pkgname}
-  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+  cargo fetch --locked --target "`uname -m`-unknown-linux-gnu"
 }
 
 build() {
   cd ${pkgname}
-  cargo build --frozen --release --all-features
+  cargo build --release --all-features
 }
 
 check() {
