diff --git a/PKGBUILD b/PKGBUILD
index ff3d67f..42ad386 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -15,7 +15,7 @@ depends=('imlib2' 'libx11'                 # core dependencies
          'hicolor-icon-theme')             # make icon
 
 source=("$pkgname-$pkgver.tar.gz"::"https://codeberg.org/nsxiv/nsxiv/archive/v$pkgver.tar.gz")
-sha256sums=('49ef1eb775ef6c34f55dada7a3f446c9c5c6773c9e208509ffef27a656338a90')
+sha256sums=('09d1d72b3cbcf17a04e26beb5e81acc9495aaba1f8f1be907bdcd8e4e3007db3')
 
 prepare() {
   cd "$pkgname"
