diff --git a/PKGBUILD b/PKGBUILD
index f89450f..0022f7f 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -10,17 +10,18 @@ arch=('x86_64')
 url="https://www.coker.com.au/bonnie++/"
 license=('GPL')
 depends=('perl')
+options=(debug)
 source=(https://www.coker.com.au/$pkgname/$pkgname-$pkgver.tgz)
 sha256sums=('a8d33bbd81bc7eb559ce5bf6e584b9b53faea39ccfb4ae92e58f27257e468f0e')
 build() {
   cd "$srcdir"/${pkgname}-${pkgver}
 
-  sed \
-    -e "s:CFLAGS=-O2 @debug@ -DNDEBUG \$(WFLAGS) \$(MORECFLAGS):CFLAGS=${CFLAGS} -std=c++14:" \
-    -e "s:LINK=@CXX@:LINK=@CXX@ ${LDFLAGS}:" \
-    -i Makefile.in
   ./configure --prefix=/usr
-  make -Otarget V=1
+  sed \
+    -e "s:CFLAGS=-O2  -DNDEBUG \$(WFLAGS) \$(MORECFLAGS):CFLAGS=-std=c++14 ${CFLAGS}:" \
+    -e "s:LINK=g++:LINK=g++ ${LDFLAGS}:" \
+    -i Makefile
+  make V=1
 }
 
 package() {
