diff --git a/PKGBUILD b/PKGBUILD
index 74952d5..51895a6 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -103,7 +103,7 @@ build() {
     -D b_ndebug=true
     -D dri3=enabled
     -D egl=enabled
-    -D gallium-drivers=r300,r600,radeonsi,nouveau,virgl,svga,swrast,i915,iris,crocus,zink,d3d12
+    -D gallium-drivers=r300,r600,radeonsi,nouveau,virgl,svga,swrast,zink,d3d12
     -D gallium-extra-hud=true
     -D gallium-nine=true
     -D gallium-omx=bellagio
@@ -128,7 +128,7 @@ build() {
     -D shared-glapi=enabled
     -D valgrind=enabled
     -D video-codecs=vc1dec,h264dec,h264enc,h265dec,h265enc
-    -D vulkan-drivers=amd,intel,intel_hasvk,swrast,virtio
+    -D vulkan-drivers=amd,intel,swrast,virtio
     -D vulkan-layers=device-select,intel-nullhw,overlay
   )
 
