diff --git a/PKGBUILD b/PKGBUILD
index 0fff224..87455db 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -14,9 +14,9 @@ sha512sums=('7fe1c2fcc5de06b89f4a5c182baeb0a98f0328ae6bf03a79e76c4af7ee4d86cd8f6
 
 prepare() {
   cd "$srcdir"/$pkgname-$pkgver
-#  XSLVER=$(pacman -Q docbook-xsl | cut -d' ' -f2 | cut -d- -f1)
-#  sed -i "s|/usr/share/sgml/docbook/stylesheet/xsl/nwalsh|/usr/share/xml/docbook/xsl-stylesheets-$XSLVER|g" configure.ac
-#  sed -i "s| *<|<|g" stalonetray.xml.in
+  XSLVER=$(pacman -Q docbook-xsl | cut -d' ' -f2 | cut -d- -f1)
+  sed -i "s|/usr/share/sgml/docbook/stylesheet/xsl/nwalsh|/usr/share/xml/docbook/xsl-stylesheets-$XSLVER|g" configure.ac
+  sed -i "s| *<|<|g" stalonetray.xml.in
 }
 
 build() {
