diff --git a/PKGBUILD b/PKGBUILD
index e47c568..046a0e4 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -5,7 +5,7 @@ pkgname=perl-mail-box-parser-c
 pkgver=3.010
 pkgrel=4
 pkgdesc="Perl/CPAN Module Mail::Box::Parser::C : extends MailBox with the speed of C"
-arch=("x86_64")
+arch=("loong64" "x86_64")
 url="https://search.cpan.org/dist/Mail-Box-Parser-C"
 license=("GPL" "PerlArtistic")
 options=('!emptydirs')
