diff --git a/PKGBUILD b/PKGBUILD
index be66ac9..2aa0960 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -19,12 +19,12 @@ b2sums=('fa4259a553528484c8a2d959fff4eac6a608cf4c22c68f4c55326310dd11530c864a13e
 
 prepare() {
   cd shotgun-$pkgver
-  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+  cargo fetch --locked --target "`uname -m`-unknown-linux-gnu"
 }
 
 build() {
   cd shotgun-$pkgver
-  cargo build --release --locked --offline
+  cargo build --release
 }
 
 package() {
