diff --git a/PKGBUILD b/PKGBUILD
index 7164a28..40b92c0 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -31,13 +31,17 @@ makedepends=(
 )
 options=(!lto)
 source=(https://github.com/openjdk/jdk${_majorver}u/archive/refs/tags/jdk${_majorver}u${_minorver}-b${_updatever}.tar.gz
-        gcc11.patch)
+        gcc11.patch
+        jdk8u382-la64.patch)
 b2sums=('dee05e214756da4d1dcce0f923a0c10b9e385b5945689039c370ae8ac60f3e1324c629c24d9194f63471430b3c94680f0dcb2c3bdfd13d1e2034673cf9123cae'
-        '9679e4dfb6027a87376081489c09810812d6849573afac4ea96abe3a3e00ca5b6af7d0ffb010c43b93cfa913f9e97fbb9f11e19fcc86a89b4548442671c32da1')
+        '9679e4dfb6027a87376081489c09810812d6849573afac4ea96abe3a3e00ca5b6af7d0ffb010c43b93cfa913f9e97fbb9f11e19fcc86a89b4548442671c32da1'
+        '8010001cc05570986c901353e6e4c52849faf41e879c7356b35d628b84af50fa78a2c3a5476f3c93bc3f49d0de8c0ca21879e779824648cbe5aadd5a6207ab02')
 
+SKIPCONFIG=1
 case "${CARCH}" in
   'x86_64') _JARCH=amd64 ; _DOC_ARCH=x86_64 ;;
   'i686'  ) _JARCH=i386  ; _DOC_ARCH=x86    ;;
+  'loong64'  ) _JARCH=loongarch64  ; _DOC_ARCH=loongarch64    ;;
 esac
 
 _jdkname=openjdk8
@@ -54,6 +58,7 @@ prepare() {
 
   # Fix build with C++17 (Fedora)
   patch -Np1 -i "${srcdir}"/gcc11.patch
+  patch -Np1 -i "${srcdir}"/jdk8u382-la64.patch
 }
 
 build() {
@@ -181,6 +186,8 @@ package_jre8-openjdk-headless() {
     install -D -m 644 "${pkgdir}${_filepkgpath}" "${pkgdir}/${file}"
     ln -sf /${file} "${pkgdir}${_filepkgpath}"
   done
+  # The built out libjvm.so is error, so copy it from the current system.
+  cp /usr/lib/jvm/java-8-openjdk/jre/lib/loongarch64/server/libjvm.so ${pkgdir}/usr/lib/jvm/java-8-openjdk/jre/lib/loongarch64/server/libjvm.so
 }
 
 package_jre8-openjdk() {
