diff --git a/PKGBUILD b/PKGBUILD
index 840faa5..881fa91 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -6,7 +6,7 @@ pkgname=just
 pkgver=1.23.0
 pkgrel=1
 pkgdesc="A handy way to save and run project-specific commands"
-arch=("x86_64")
+arch=("loong64" "x86_64")
 url="https://github.com/casey/just"
 license=("custom:CC0")
 depends=("gcc-libs")
@@ -21,7 +21,13 @@ prepare() {
 
 build() {
   cd "${pkgname}-${pkgver}"
-  cargo build --frozen --release
+  find -name Cargo.lock -exec rm -f {} \;
+  mkdir -p .cargo
+  cat > .cargo/config.toml <<EOF
+[source.crates-io]
+registry = "https://gitee.com/yetist/crates.io-index"
+EOF
+  cargo build --release
 }
 
 check() {
