diff --git a/PKGBUILD b/PKGBUILD
index 3339ce3..3dd2170 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -22,13 +22,20 @@ validpgpkeys=("64B13F7117D6E07D661BBCE0FE763A64F5E54FD6")
 
 prepare() {
   cd "${pkgname}-${pkgver}"
-  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+  mkdir .cargo
+  cat > .cargo/config.toml <<EOF
+[source]
+[source.crates-io]
+registry = "https://gitee.com/yetist/crates.io-index"
+EOF
+  cargo update
+  cargo fetch --target "`uname -m`-unknown-linux-gnu"
 }
 
 build() {
   cd "${pkgname}-${pkgver}"
   CFLAGS+=' -ffat-lto-objects'
-  cargo build --frozen --release
+  cargo build --release
   make docs
 }
 
