diff --git a/PKGBUILD b/PKGBUILD
index eaa60f4..5705f83 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -8,7 +8,7 @@ arch=('any')
 url='https://www.bouncycastle.org/java.html'
 license=('MIT')
 depends=('java-runtime-headless')
-makedepends=('ant' 'strip-nondeterminism')
+makedepends=('git' 'ant' 'strip-nondeterminism')
 source=("$pkgname-$pkgver.tar.gz::https://github.com/bcgit/bc-java/archive/refs/tags/r${pkgver/./rv}.tar.gz")
 sha512sums=('7d2abab42a7e29159ae063244a4296708d1269e5a8250f0e2f62b095916d509e6e4213f4b32d45f375e1aabea572860d9b81df9ee5efcdff9b9e569864c9f8db')
 
