diff --git a/PKGBUILD b/PKGBUILD
index 99de9ed..bfa9a80 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -36,6 +36,13 @@ prepare() {
 
 build() {
 	cd "$_archive"
+	pushd ruby
+	gem-2.7 sources -r https://rubygems.org/
+	gem-2.7 sources -a https://rubygems.loongnix.cn
+	gem-2.7 sources -c
+#  bundle-2.7 config force_ruby_platform true # build from sources as some prebuilt gems are not available for newer ruby
+#  bundle-2.7 install --path vendor/bundle
+	popd
 	make V=1 BUILD_TAGS="tracer_static tracer_static_jaeger"
 }
 
