diff --git a/PKGBUILD b/PKGBUILD
index 18e1775..a7232a9 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -2,7 +2,7 @@
 # Maintainer: Bartłomiej Piotrowski <bpiotrowski@archlinux.org>
 
 pkgname=archlinux-keyring
-_tag='5857f33114c78e3f03fb250c5ce2c12259609ce0' # git rev-parse ${pkgver}
+_tag='c1ddf69bbd0079132716d13307c1bb38b9b27079' # git rev-parse ${pkgver}
 pkgver=20231222
 pkgrel=1
 pkgdesc='Arch Linux PGP keyring'
@@ -13,7 +13,7 @@ install=$pkgname.install
 depends=('pacman')
 makedepends=('git' 'python' 'sequoia-sq' 'pkgconf' 'systemd')
 checkdepends=('python-coverage' 'python-pytest')
-source=("archlinux-keyring::git+https://gitlab.archlinux.org/archlinux/archlinux-keyring.git#tag=${_tag}?signed")
+source=("archlinux-keyring::git+https://github.com/loongarchlinux/archlinux-keyring.git#tag=${_tag}?signed")
 sha256sums=('SKIP')
 validpgpkeys=('02FD1C7A934E614545849F19A6234074498E9CEE'  # Christian Hesse <eworm@archlinux.org>
               'C7E7849466FE2358343588377258734B41C31549'  # David Runge <dvzrv@archlinux.org>
