diff --git a/PKGBUILD b/PKGBUILD
index 47e412c..90924e7 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -11,11 +11,10 @@ license=('LGPL')
 depends=('hunspell' 'sqlite' 'libedit')
 makedepends=('python' 'swig' 'apache-ant' 'java-environment=11')
 options=('!makeflags')
-source=(https://www.abisource.com/downloads/${pkgname}/${pkgver}/${pkgname}-${pkgver}.tar.gz{,.asc})
+source=(https://github.com/opencog/link-grammar/archive/refs/tags/${pkgname}-${pkgver}.tar.gz)
 validpgpkeys=('6407453C98BECC19ADB03D82EB6AA534E0C0651C'
               '8305252160000B5E89843F5464A99A8201045933') # Linas Vepstas <linasvepstas@gmail.com>
-sha256sums=('ef89a322f943607546a667ecb4fdf31d56cab5ed11c856873ac72fa3d352f7a3'
-            'SKIP')
+sha256sums=('e0cd1b94cc9af20e5bd9a04604a714e11efe21ae5e453b639cdac050b6ac4150')
 
 build() {
   cd ${pkgname}-${pkgver}
