diff --git a/PKGBUILD b/PKGBUILD
index 8b854d7..417dc24 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -70,8 +70,9 @@ package() {
     "$pkgdir/usr/lib/systemd/system/distccd.service"
 
   # Package symlinks
-  _targets=(c++ c89 c99 cc clang clang++ cpp g++ gcc $CARCH-pc-linux-gnu-g++
-            $CARCH-pc-linux-gnu-gcc $CARCH-pc-linux-gnu-gcc-$(gcc -dumpversion))
+  ARCH=`uname -m`
+  _targets=(c++ c89 c99 cc clang clang++ cpp g++ gcc $ARCH-pc-linux-gnu-g++
+            $ARCH-pc-linux-gnu-gcc $ARCH-pc-linux-gnu-gcc-$(gcc -dumpversion))
   install -d "$pkgdir/usr/lib/$pkgname/bin"
   for bin in "${_targets[@]}"; do
     # For whitelist since version 3.3, ref FS#57978
