diff --git a/PKGBUILD b/PKGBUILD
index 1ca1e5c..9654947 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -125,11 +125,11 @@ check() {
   # check if VideoReader is build
   # VideoReader depends on ffmpeg
   cd "${srcdir}/python-${_pkgname}-${pkgver}"
-  PYTHONPATH="${PWD}/build/lib.linux-${CARCH}-cpython-${python_version}" \
+  PYTHONPATH="${PWD}/build/lib.linux-`uname -m`-cpython-${python_version}" \
   python -c "from torchvision.io import VideoReader"
 
   cd "${srcdir}/python-${_pkgname}-cuda-${pkgver}"
-  PYTHONPATH="${PWD}/build/lib.linux-${CARCH}-cpython-${python_version}" \
+  PYTHONPATH="${PWD}/build/lib.linux-`uname -m`-cpython-${python_version}" \
   python -c "from torchvision.io import VideoReader"
 }
 
