diff --git a/PKGBUILD b/PKGBUILD
index 52e38fd..6097887 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -21,7 +21,7 @@ makedepends=(lua
              luarocks)
 options=(debug)
 _archive="$pkgbase-$pkgver"
-_rock="$_archive-$_rockrel.linux-$CARCH.rock"
+_rock="$_archive-$_rockrel.linux-`uname -m`.rock"
 _rockspec="$_archive-$_rockrel.rockspec"
 source=("$url/archive/$pkgver/$_archive.tar.gz")
 sha512sums=('45834b02d863a65c6de2ad3c5c671935ae512dd906233a647269438aeeb37c7bbd6e7166aefd8da72480b33e1aa79a86ed070ed8800a8e0f303ccc74918f3bb8')
