diff --git a/PKGBUILD b/PKGBUILD
index a3dc23d..f19fe96 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -5,7 +5,7 @@ pkgname=perl-mail-transport-dbx
 pkgver=0.07
 pkgrel=22
 pkgdesc="Perl/CPAN Module Mail::Transport::Dbx : Parse Outlook Express mailboxes"
-arch=("x86_64")
+arch=("loong64" "x86_64")
 url="https://search.cpan.org/dist/Mail-Transport-Dbx"
 license=("GPL" "PerlArtistic")
 options=('!emptydirs')
