diff --git a/PKGBUILD b/PKGBUILD
index 2cdf3da..aaa1acd 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -10,7 +10,7 @@ arch=(x86_64)
 license=(GPL2)
 depends=(bash glibc libyaml python)
 makedepends=(meson swig python-setuptools-scm)
-checkdepends=(valgrind)
+makedepends+=(valgrind)
 source=(https://www.kernel.org/pub/software/utils/dtc/dtc-$pkgver.tar.xz
         fix_test.patch::https://github.com/dgibson/dtc/commit/32174a66efa4ad19fc6a2a6422e4af2ae4f055cb.patch
 	fix_meson_version.patch::https://github.com/dgibson/dtc/commit/64a907f08b9bedd89833c1eee674148cff2343c6.patch
