diff --git a/PKGBUILD b/PKGBUILD
index df867bf..0cadfab 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -14,10 +14,12 @@ depends=('glibc')
 provides=('libcrypto.so' 'libssl.so')
 makedepends=('perl')
 source=("https://www.openssl.org/source/${_pkgname}-${_ver}.tar.gz"{,.asc}
-        'ca-dir.patch')
+        'ca-dir.patch'
+        'openssl-la64.patch')
 sha256sums=('cf3098950cb4d853ad95c0841f1f9c6d3dc102dccfcacd521d93925208b76ac8'
             'SKIP'
-            '75aa8c2c638c8a3ebfd9fa146fc61c7ff878fc997dc6aa10d39e4b2415d669b2')
+            '75aa8c2c638c8a3ebfd9fa146fc61c7ff878fc997dc6aa10d39e4b2415d669b2'
+            '2b146d69cfe81801012da66a0880fe170c7bd31f03c4b045e31e9f22807b97e8')
 validpgpkeys=('8657ABB260F056B1E5190839D9C4D26D0E604491'
               '7953AC1FBC3DC8B3B292393ED5E9E43F7DF9EE8C'
               'A21FAB74B0088AA361152586B8EF1A6BA9DA2D5C'
@@ -28,6 +30,7 @@ prepare() {
 
 	# set ca dir to /etc/ssl by default
 	patch -p0 -i "$srcdir/ca-dir.patch"
+	patch -p1 -i "$srcdir/openssl-la64.patch"
 }
 
 build() {
@@ -35,7 +38,7 @@ build() {
 
 	# mark stack as non-executable: http://bugs.archlinux.org/task/12434
 	./Configure --prefix=/usr --openssldir=/etc/ssl --libdir=lib/openssl-1.1 \
-		shared no-ssl3-method enable-ec_nistp_64_gcc_128 linux-x86_64 \
+		shared no-ssl3-method enable-ec_nistp_64_gcc_128 linux-`uname -m` \
 		"-Wa,--noexecstack ${CPPFLAGS} ${CFLAGS} ${LDFLAGS}"
 
 	make depend
