diff --git a/PKGBUILD b/PKGBUILD
index f3dc711..4c00e59 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -8,7 +8,7 @@ arch=('any')
 url="https://github.com/pycontribs/wstools"
 license=('custom')
 depends=('python-six')
-makedepends=('python-pbr' 'python-setuptools')
+makedepends=('python-pbr' 'python-setuptools' 'python-pip')
 checkdepends=('python-pytest-runner' 'autopep8' 'python-pytest-cov')
 source=("https://github.com/pycontribs/wstools/archive/$pkgver/$pkgname-$pkgver.tar.gz"
         python310.patch)
