diff --git a/PKGBUILD b/PKGBUILD
index 07f855a..50d3360 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -17,8 +17,10 @@ makedepends=(
 )
 groups=(gnome-extra)
 _commit=53db1397d6a711862023d7a3070e785235a98c87  # tags/44.0^0
-source=("git+https://gitlab.gnome.org/GNOME/hitori.git#commit=$_commit")
-sha256sums=('SKIP')
+source=("git+https://gitlab.gnome.org/GNOME/hitori.git#commit=$_commit"
+        hitori-fix-meson.patch)
+sha256sums=('SKIP'
+            '20b67731b04efadbc3f5b03958a25eddacc17c05c842943a4cd99fe37fd0d503')
 
 pkgver() {
   cd hitori
@@ -27,6 +29,7 @@ pkgver() {
 
 prepare() {
   cd hitori
+  patch -p1 -i $srcdir/hitori-fix-meson.patch
 }
 
 build() {
