diff --git a/PKGBUILD b/PKGBUILD
index 6378eca..0c36d57 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -15,13 +15,13 @@ options=('!lto')
 
 prepare() {
 	cd "SpicyLauncher-$pkgver"
-	cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+	cargo fetch --locked --target "`uname -m`-unknown-linux-gnu"
 }
 
 build() {
 	cd "SpicyLauncher-$pkgver/gui"
 	yarn install --ignore-engines
-	yarn tauri build --target "$CARCH-unknown-linux-gnu" --bundles none
+	yarn tauri build --target "`uname -m`-unknown-linux-gnu" --bundles none
 	cd ..
 	cargo build --release --frozen
 }
