diff --git a/PKGBUILD b/PKGBUILD
index 8de8ee1..fa99bcd 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -11,12 +11,12 @@ pkgdesc="Collection of user space tools for general SMBus access and hardware mo
 arch=('x86_64')
 url="https://hwmon.wiki.kernel.org/lm_sensors"
 license=('GPL' 'LGPL')
-depends=('bash' 'glibc')
-makedepends=('rrdtool' 'perl' 'git')
 optdepends=('rrdtool: for logging with sensord'
             'perl: for sensor detection and configuration convert')
+makedepends=('rrdtool' 'perl' 'git')
 provides=('libsensors.so')
 backup=('etc/sensors3.conf' 'etc/healthd.conf' 'etc/conf.d/sensord')
+options=('debug')
 #source=(lm_sensors-${pkgver}.tar.gz::https://github.com/lm-sensors/lm-sensors/archive/V${_pkgver}.tar.gz
 source=("git+https://github.com/groeck/lm-sensors.git#commit=${_commit}"
 	healthd healthd.conf healthd.service sensord.conf)
