diff --git a/PKGBUILD b/PKGBUILD
index 629c597..6996e13 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -23,7 +23,7 @@ options=('staticlibs')
 source=(${pkgname}-${pkgver}.tar.gz::https://github.com/KhronosGroup/glslang/archive/${pkgver}.tar.gz
         ${pkgname}-3420.patch::https://github.com/KhronosGroup/glslang/pull/3420.patch)
 sha256sums=('1c4d0a5a38c8aaf89a2d7e6093be734320599f5a6775b2726beeb05b0c054e66'
-            '8930d3829bae4e0cd911bf63728d4d49d736d261af1e2cef912b769e6fa8373f')
+            '785a72531afe9dc17c88eb24a86c7137e15560f1dc4389aa811c5b0f4979029f')
 
 prepare() {
   cd ${pkgname}-${pkgver}
