diff --git a/PKGBUILD b/PKGBUILD
index ff4d285..eff7407 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -19,8 +19,8 @@ source=("https://dl.discordapp.net/apps/linux/$pkgver/$pkgname-$pkgver.tar.gz"
         "LICENSE-$pkgver.html::https://discordapp.com/terms"
         "OSS-LICENSES-$pkgver.html::https://discordapp.com/licenses")
 sha512sums=('ec11acfe3f96762cd7c230eb38e098818d9ebf32ff530f061ed3dbb5d08c3303bb4d20206510e3621f21ba89c08474367c13e7fd3c46f80b7229db37fb81db6c'
-            'a47ee83d4878f936a42faaa97d87f8a23261be1d5a5fc1c072be6b8a1fd505ccb7a3b39a8f7675677a241421f80b4cc57957d32eaf1fee96f91e0fe976cce167'
-            '4587af901dc20a843f05ba54411c7d9a39d6976e4c3689fb94c504d6ed8af9dd46aa54c2b87a9dcdee1dd4bcbf79d68be66556ce9d329959f849cde2a8522e4a')
+            '6fd75707e5ddf52454b0172309a820def89a0d400336bbe40c9176eaebcf6d65d2d083636c76a2595a14193508254520007bc8d31e14f4be929334a5f3bf91bb'
+            'c1de9d8abf32a750c27c44232b1f1f8f8b569c1a16fd081587efc5a8bef8a124b278e333f9cdd62341f4e2fdc60fb7a0e7d7d13213315feee6058f4a225258ba')
 
 prepare() {
   cd $_pkgname
