diff --git a/PKGBUILD b/PKGBUILD
index 19fe357..7e01fff 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -14,8 +14,9 @@ sha256sums=('7ca3a04bb4ab83387538235decc42a535097a05d2fb9f2266d0c47b33119501f')
 
 build() {
   cd "$pkgname-$pkgver"
-  export GOFLAGS="-buildmode=pie -mod=vendor -trimpath"
+  export GOFLAGS="-mod=vendor -trimpath"
   export CGO_LDFLAGS="$LDFLAGS"
+  export GOPROXY=https://goproxy.cn
   go build -o go-md2man .
   ./go-md2man -in=go-md2man.1.md -out=go-md2man.1
 }
