diff --git a/PKGBUILD b/PKGBUILD
index 9466bd9..817bec3 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -10,7 +10,7 @@ _kubeone_commit="1195366fd0cf11f314d194a3b29b6a782afde9a8"
 pkgrel=1
 pkgdesc="A lifecycle management tool for Highly-Available Kubernetes clusters"
 url="https://github.com/kubermatic/kubeone"
-arch=("x86_64")
+arch=("loong64" "x86_64")
 license=("Apache")
 makedepends=("go" "git")
 optdepends=("terraform: sourcing data about infrastructure and control plane nodes")
