diff --git a/PKGBUILD b/PKGBUILD
index ab2b391..a41b797 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -62,7 +62,10 @@ source=("git+https://github.com/electron/electron.git#tag=v$pkgver"
         std-vector-non-const.patch
         use-system-libraries-in-node.patch
         libxml2-2.12.patch
-        icu-74.patch)
+        icu-74.patch
+#electron-la64.patch
+       )
+# shellcheck disable=SC2034
 sha256sums=('SKIP'
             'SKIP'
             'SKIP'
@@ -77,6 +80,7 @@ sha256sums=('SKIP'
             'ff588a8a4fd2f79eb8a4f11cf1aa151298ffb895be566c57cc355d47f161f53f'
             'bfae9e773edfd0ddbc617777fdd4c0609cba2b048be7afe40f97768e4eb6117e'
             '547e092f6a20ebd15e486b31111145bc94b8709ec230da89c591963001378845')
+#            '671fd958b429414a66c209c8b91b6876a77bf4ed38244044ba14703de3f02a66')
 
 # Possible replacements are listed in build/linux/unbundle/replace_gn_files.py
 # Keys are the names in the above script; values are the dependencies in Arch
@@ -145,6 +149,7 @@ EOF
 
   pushd src/electron
   patch -Np1 -i ../../std-vector-non-const.patch
+#  patch -Np1 -i ../../electron-la64.patch
   popd
 
   echo "Running hooks..."
