diff --git a/PKGBUILD b/PKGBUILD
index f11d2ae..5eda0f9 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -25,8 +25,12 @@ build() {
     export CGO_CXXFLAGS="${CXXFLAGS}"
     export CGO_LDFLAGS="${LDFLAGS}"
     export GOFLAGS="-buildmode=pie -trimpath -mod=readonly -modcacherw"
+    export GOPROXY=https://goproxy.cn
 
     cd "${pkgname}-${pkgver}"
+    go mod edit -replace=go.etcd.io/bbolt=go.etcd.io/bbolt@v1.3.7-0.20221114114133-eedea6cb26ef
+    go mod edit -replace=github.com/u-root/uio=github.com/loongarch64/uio@dev-main
+    go mod tidy
     go build -trimpath -ldflags "-X github.com/Dreamacro/clash/constant.Version=${pkgver}" -mod=readonly
 }
 
