diff --git a/PKGBUILD b/PKGBUILD
index ccffbe5..e24a1ba 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -21,12 +21,12 @@ b2sums=('f5b8a1b1f10b42ddb98c7ea400a062715e3ab9c2023adece88052126847ab992db52258
 
 prepare() {
   cd $pkgname-$pkgver
-  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+  cargo fetch --locked --target "`uname -m`-unknown-linux-gnu"
 }
 
 build() {
   cd $pkgname-$pkgver
-  cargo build --release --locked --offline --all-features
+  cargo build --release --offline --all-features
 }
 
 package() {
