diff --git a/PKGBUILD b/PKGBUILD
index 57014d3..fe2d68c 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -17,10 +17,12 @@ validpgpkeys=('177F4010FE56CA3336300305F1656F24C74CD1D8') # MariaDB Signing Key
 # rsync source via https and hope it does not hurt them too much.
 # https://mariadb.com/kb/en/library/mirror-sites-for-mariadb/
 source=("https://rsync.osuosl.org/pub/mariadb/mariadb-${pkgver}/source/mariadb-${pkgver}.tar.gz"{,.asc}
-        '0001-arch-specific.patch')
+        '0001-arch-specific.patch'
+	'mariadb-fix-build.patch')
 sha256sums=('faedbd8790c7ee65b348c0169706b4bae91eb6ce7335a76b27dbd8813c42e21b'
             'SKIP'
-            '3289efb3452d199aec872115f35da3f1d6fd4ce774615076690e9bc8afae1460')
+            '3289efb3452d199aec872115f35da3f1d6fd4ce774615076690e9bc8afae1460'
+            'ec16ca91395e1d49b7586f8a91b74039649d330f1b675e808858a2320bce10b5')
 
 prepare() {
   cd $pkgbase-$pkgver/
@@ -30,6 +32,8 @@ prepare() {
   #  * force preloading jemalloc for memory management
   #  * make systemd-tmpfiles create MYSQL_DATADIR
   patch -Np1 < ../0001-arch-specific.patch
+
+  patch -Np1 < ../mariadb-fix-build.patch
 }
 
 build() {
