diff --git a/PKGBUILD b/PKGBUILD
index 9b13437..2be58f6 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -12,7 +12,7 @@ url="https://github.com/nextcloud/deck"
 license=(AGPL-3.0-or-later)
 groups=(nextcloud-apps)
 makedepends=(nextcloud)
-checkdepends=(yq)
+makedepends+=(yq)
 source=(https://github.com/nextcloud-releases/$_name/releases/download/v$pkgver/$_name-v$pkgver.tar.gz)
 sha512sums=('e1ffbbca8a86d3657b09088d66b090103627df108abceaec64c6e5f806c04d498a77cdecab8ec0d6cb273c859d3ff60388dd2dda0f7adc9a234016df5c9d8943')
 b2sums=('49d905f9c5f7741e5150e2641e3bd342a01c3a22b95233fcc129dfd96d84a6d00849d4912748a72f96f4820641f1b777a0b25da812a6906aadef5aafe7e7c96b')
