diff --git a/PKGBUILD b/PKGBUILD
index 2ea330c..9383d6b 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -19,6 +19,7 @@ depends=(
 makedepends=(
   clang
   git
+  gettext
 )
 _commit=ed1e5f514d2e314ceacde841da485aa69cccba4a
 source=(libretro-genesis-plus-gx::git+https://github.com/libretro/Genesis-Plus-GX.git#commit=${_commit})
@@ -31,6 +32,10 @@ pkgver() {
 
 build() {
   export CC=clang # FS#71188 - [libretro-genesis-plus-gx] segfaults when compiled with -O2
+  CFLAGS=${CFLAGS/-mlsx /}
+  CXXFLAGS=${CXXFLAGS/-mlsx /}
+  CFLAGS=${CFLAGS/-fstack-clash-protection/}
+  CXXFLAGS=${CXXFLAGS/-fstack-clash-protection/}
   make \
     SHARED_LIBVORBIS=1 \
     SYSTEM_ZLIB=1 \
