diff --git a/PKGBUILD b/PKGBUILD
index c2492a3..a9a677f 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -37,11 +37,13 @@ source=(
   'no-python-pip.patch'
   'no-recompile-on-install.patch'
   'no-amdgpu-manpage.patch'
+  2183.patch
 )
 b2sums=('SKIP'
         'd83da0ce0222c1aea1fc0c97bbf8a40f3cd5a6b5d55ee973b64f97bd9769df265b148e89cee8ee6564f065adc00552b511904f322555ac659b735933d42a9a64'
         'e4b7c4831fa513d602c73e377847705240a6a42ee1986effd10a589784bd0ad818032ff8283c1f9fd17cb7ddf3204e4a932796a1df816afc30a0e594c92b50f6'
-        '9c713724e8f6b062f7a09e34555d31e5aa0315db6308b7527835484eaad8dbf5deac5c66521bf5a819462d5f38c64f6602ba421f7bbb73180a3b05189816c8f6')
+        '9c713724e8f6b062f7a09e34555d31e5aa0315db6308b7527835484eaad8dbf5deac5c66521bf5a819462d5f38c64f6602ba421f7bbb73180a3b05189816c8f6'
+        'b0d762bb3cc93608fe32f96d3c018ada1e8708196031cb2f80ec6350af1d96d121476288083e1053805dcdeaf9fa7e636e08142a5ea83ecaa393341cffc9c664')
 
 pkgver() {
   cd "$pkgname"
@@ -60,6 +62,7 @@ prepare() {
 
   # do not install amdgpu_plugin manpage
   patch -p1 -i "$srcdir/no-amdgpu-manpage.patch"
+  patch -p1 -i "$srcdir/2183.patch"
 }
 
 build() {
