diff --git a/PKGBUILD b/PKGBUILD
index 4fd0e53..cfb0b59 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -15,7 +15,7 @@ license=(
   LGPL-2.0-only
   LGPL-3.0-only
 )
-makedepends=(acpica inetutils python)
+makedepends=(acpica inetutils python x86_64-linux-gnu-gcc)
 options=(!makeflags !strip)
 source=(
   seabios-$pkgver.tar.gz::https://github.com/coreboot/seabios/archive/rel-$pkgver.tar.gz
@@ -73,15 +73,18 @@ _build_bios() {
   local build_target=$4
 
   echo "Building target with config $config, output_name $output_name, binary_name $binary_name and build_target $build_target..."
+  CFLAGS=${CFLAGS/-mabi=lp64d -march=la464/}
+  CXXFLAGS=${CXXFLAGS/-mabi=lp64d -march=la464/}
 
   make clean distclean -C $pkgbase-rel-$pkgver
 
   cp -v $config $pkgbase-rel-$pkgver/.config
   # NOTE: refer to $pkgbase-rel-$pkgver/src/config.h for explanation of debug levels
   echo "CONFIG_DEBUG_LEVEL=$_debug_level" >> $pkgbase-rel-$pkgver/.config
-  make oldnoconfig V=1 -C $pkgbase-rel-$pkgver
+  make oldnoconfig V=1 -C $pkgbase-rel-$pkgver \
+      CROSS_PREFIX=x86_64-linux-gnu-
 
-  make -C $pkgbase-rel-$pkgver V=1 EXTRAVERSION=-$pkgrel PYTHON=python3 $build_target
+  make -C $pkgbase-rel-$pkgver V=1 EXTRAVERSION=-$pkgrel PYTHON=python3 $build_target CROSS_PREFIX=x86_64-linux-gnu-
 
   cp $pkgbase-rel-$pkgver/out/$output_name output/$binary_name
 }
