diff --git a/PKGBUILD b/PKGBUILD
index a0dc899..ffc67e1 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -11,7 +11,8 @@ depends=(gcc-libs)
 makedepends=(git)
 provides=(libwebrtc_audio_processing.so)
 _commit=e882a5442ac22c93648e12837248d651d18b9247  # tags/v0.3.1^0
-source=("git+https://gitlab.freedesktop.org/pulseaudio/webrtc-audio-processing.git#commit=$_commit")
+source=("git+https://gitlab.freedesktop.org/pulseaudio/webrtc-audio-processing.git#commit=$_commit"
+        webrtc-audio-processing-la64.patch)
 b2sums=('SKIP')
 
 pkgver() {
@@ -22,6 +23,7 @@ pkgver() {
 prepare() {
   cd $pkgname
   NOCONFIGURE=1 ./autogen.sh
+  patch -p1 -i $srcdir/webrtc-audio-processing-la64.patch
 }
 
 build() {
