diff --git a/PKGBUILD b/PKGBUILD
index 4b1b2d4..afdf4fe 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -58,6 +58,7 @@ source=(${_mirror}/libreoffice{,-help,-translations}-${_LOver}.tar.xz{,.asc}
 	${_additional_source_url2}/odfvalidator-1.2.0-incubating-SNAPSHOT-jar-with-dependencies-971c54fd38a968f5860014b44301872706f9e540.jar # for test suite
 	${_additional_source_url2}/f543e6e2d7275557a839a164941c0a86e5f2c3f2a0042bfc434c88c6dde9e140-opens___.ttf
 	${_additional_source_url2}/185d60944ea767075d27247c3162b3bc-unowinreg.dll
+	https://gitweb.gentoo.org/repo/gentoo.git/plain/app-office/libreoffice/files/libreoffice-7.6.5.2-gcc14.patch
 	make-pyuno-work-with-system-wide-module-install.diff
         623ea5c.diff
         fix-build-against-system-libxml-2.12.diff
@@ -110,6 +111,7 @@ sha256sums=('13fea7b8f24c776313b9e08628aa590390bea45064be73bc70ee7b1b70aa6a1e'
             '984f2a479df79e27e7b01a5815ac53ae64e07746b882262d8a64566494515504'
             'f543e6e2d7275557a839a164941c0a86e5f2c3f2a0042bfc434c88c6dde9e140'
             'eafde646a7dbe46d20c291685b0beac2382174d78d66ee990e229a1bf6e6cec6'
+            '17cc24b1d4d47562ea28bd89b302d40c1e112494a1a6b52c64e3a5292c18453f'
             'c463654a73ecfbc242ff109726fb4faecdbfb3d91affafe919b24bea65afb563'
             '440c9af5f3d1213d8ed7177282380f25cbc981cabc8b590dcb777aaae84178e5'
             '793a52abff29b3db51a1db9686b561911b9b3de70bd6dd02bbc1d78fcd960648'
@@ -145,6 +147,7 @@ prepare() {
 
         # fix build with icu 74
         patch -Np1 -i "${srcdir}"/libreoffice-7.5.8.2-icu-74-compatibility.patch
+	patch -Np1 -i ${srcdir}/libreoffice-7.6.5.2-gcc14.patch
 
 	#use the CFLAGS but remove the LibO overridden ones
 	for i in $CFLAGS; do
@@ -164,6 +167,8 @@ build() {
         # Build only minimal debug info to reduce size (~1.2GB -> ~225MB)
         CFLAGS=${CFLAGS/-g /-g1 }
         CXXFLAGS=${CXXFLAGS/-g /-g1 }
+        CFLAGS=${CFLAGS/-mlsx /}
+        CXXFLAGS=${CFLAGS/-mlsx /}
 
         # this uses malloc_usable_size, which is incompatible with fortification level 3
         # /usr/lib/libreoffice/program/libskialo.so uses malloc_usable_size
@@ -223,7 +228,7 @@ build() {
 		--with-system-libmwaw \
 		--with-system-libetonyek \
 		--with-system-libfreehand \
-		--without-system-firebird \
+		--disable-firebird-sdbc \
 		--with-system-zxing \
 		--with-system-libtommath \
 		--with-system-libatomic-ops \
