diff --git a/PKGBUILD b/PKGBUILD
index 8fb47e3..6d92664 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -11,7 +11,7 @@ arch=(x86_64)
 url='https://kde.org/plasma-desktop/'
 license=(LGPL)
 depends=(plasma-workspace kunitconversion5)
-makedepends=(extra-cmake-modules qt5-webengine networkmanager-qt5)
+makedepends=(extra-cmake-modules networkmanager-qt5)
 optdepends=('purpose5: Quickshare applet'
             'quota-tools: disk quota applet'
             'qt5-webengine: dictionary and webbrowser applets'
