diff --git a/PKGBUILD b/PKGBUILD
index 89e0618..2c7763e 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -25,6 +25,11 @@ prepare() {
 
 build() {
   cd "$pkgname-$pkgver"
+  export GOPROXY=https://goproxy.cn
+  go mod edit -replace=golang.org/x/sys=github.com/golang/sys@v0.0.0-20220622161953-175b2fd9d664
+  go mod edit -replace=golang.org/x/net=github.com/golang/net@v0.0.0-20220622184535-263ec571b305
+  go mod edit -replace=go.etcd.io/bbolt=go.etcd.io/bbolt@master
+  go mod tidy
 
   export CGO_LDFLAGS="${LDFLAGS}"
   export CGO_CFLAGS="${CFLAGS}"
