diff --git a/PKGBUILD b/PKGBUILD
index dcf1016..6557602 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -10,8 +10,15 @@ url="https://www.ginac.de/CLN/"
 license=('GPL')
 depends=('gmp')
 makedepends=('texlive-latex')
-source=(https://www.ginac.de/CLN/${pkgname}-${pkgver}.tar.bz2)
-sha256sums=('f492530e8879bda529009b6033e1923c8f4aae843149fc28c667c20b094d984a')
+source=(https://www.ginac.de/CLN/${pkgname}-${pkgver}.tar.bz2
+        cln-la64.patch)
+sha256sums=('f492530e8879bda529009b6033e1923c8f4aae843149fc28c667c20b094d984a'
+            '8cea3f49ab301dff766450bfdbff3ac4e5bf64cb60307f73681267aa151bb6c5')
+
+prepare() {
+	cd "$pkgname-$pkgver"
+	patch -p1 -i "$srcdir/cln-la64.patch"
+}
 
 build() {
   cd ${pkgname}-${pkgver}
