diff --git a/PKGBUILD b/PKGBUILD
index 18df176..a55aa9b 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -72,6 +72,9 @@ build() {
 
   # provide version to build script
   echo "$pkgver" > VERSION
+  go mod edit -replace=go.etcd.io/bbolt=go.etcd.io/bbolt@v1.3.8
+  go mod edit -replace=github.com/cilium/ebpf=github.com/cilium/ebpf@v0.12.3
+  go mod tidy
 
   # set Go flags
   export CGO_CPPFLAGS="${CPPFLAGS}"
