diff --git a/PKGBUILD b/PKGBUILD
index fed85a8..75258c4 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -15,6 +15,8 @@ sha512sums=('790e3ab8dee122320ad8b3ae15f6a1cd2780222d5ae97979f614f16ba73b4b85396
 
 build() {
   cd "${srcdir}/goom2k4-0"
+  CFLAGS=${CFLAGS/-Wformat -Werror=format-security/}
+  CXXFLAGS=${CXXFLAGS/-Wformat -Werror=format-security/}
 
   ./configure --prefix=/usr \
               --without-xmms \
