diff --git a/PKGBUILD b/PKGBUILD
index 42cecaa..5cfbd4c 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -13,8 +13,10 @@ license=(GPL)
 depends=(glib2)
 makedepends=(gobject-introspection git meson appstream-glib)
 _commit=c87aac46c4893e09b1ced1cca8bb86b0a6823124  # master
-source=("git+https://gitlab.gnome.org/Archive/nautilus-sendto.git#commit=$_commit")
-sha256sums=('SKIP')
+source=("git+https://gitlab.gnome.org/Archive/nautilus-sendto.git#commit=$_commit"
+    nautilus-sendto-fix-meson.patch)
+sha256sums=('SKIP'
+            '2ba3e793e49b67866e70e65e6cd7fb8029dee62681a5dcd2d9ba1fb494a24eeb')
 
 pkgver() {
   cd $pkgname
@@ -23,6 +25,7 @@ pkgver() {
 
 prepare() {
   cd $pkgname
+  patch -p1 -i $srcdir/nautilus-sendto-fix-meson.patch
 }
 
 build() {
