diff --git a/PKGBUILD b/PKGBUILD
index 85fba0b..0f7f7dd 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -7,14 +7,14 @@
 # NOTE: valgrind requires rebuilt with each major glibc version
 
 pkgbase=glibc
-pkgname=(glibc lib32-glibc glibc-locales)
+pkgname=(glibc glibc-locales)
 pkgver=2.39
 _commit=6d1e3fb07b45e2e31e469b16cf21b24bccf8914c
 pkgrel=1
 arch=(x86_64)
 url='https://www.gnu.org/software/libc'
 license=(GPL-2.0-or-later LGPL-2.1-or-later)
-makedepends=(git gd lib32-gcc-libs python)
+makedepends=(git gd python)
 options=(staticlibs !lto)
 source=(git+https://sourceware.org/git/glibc.git#commit=${_commit}
         locale.gen.txt
@@ -32,7 +32,7 @@ b2sums=('SKIP'
         '214e995e84b342fe7b2a7704ce011b7c7fc74c2971f98eeb3b4e677b99c860addc0a7d91b8dc0f0b8be7537782ee331999e02ba48f4ccc1c331b60f27d715678')
 
 prepare() {
-  mkdir -p glibc-build lib32-glibc-build
+  mkdir -p glibc-build
 
   [[ -d glibc-$pkgver ]] && ln -s glibc-$pkgver glibc
   cd glibc
@@ -43,7 +43,7 @@ build() {
   local _configure_flags=(
       --prefix=/usr
       --with-headers=/usr/include
-      --with-bugurl=https://gitlab.archlinux.org/archlinux/packaging/packages/glibc/-/issues
+      --with-bugurl=https://github.com/loongarchlinux/core/issues
       --enable-bind-now
       --enable-fortify-source
       --enable-kernel=4.4
@@ -52,6 +52,7 @@ build() {
       --enable-systemtap
       --disable-nscd
       --disable-profile
+      --enable-obsolete-api=no
       --disable-werror
   )
 
@@ -80,24 +81,24 @@ build() {
     make info
   )
 
-  (
-    cd lib32-glibc-build
-    export CC="gcc -m32 -mstackrealign"
-    export CXX="g++ -m32 -mstackrealign"
-
-    echo "slibdir=/usr/lib32" >> configparms
-    echo "rtlddir=/usr/lib32" >> configparms
-    echo "sbindir=/usr/bin" >> configparms
-    echo "rootsbindir=/usr/bin" >> configparms
-
-    "${srcdir}"/glibc/configure \
-        --host=i686-pc-linux-gnu \
-        --libdir=/usr/lib32 \
-        --libexecdir=/usr/lib32 \
-        "${_configure_flags[@]}"
-
-    make -O
-  )
+#  (
+#    cd lib32-glibc-build
+#    export CC="gcc -m32 -mstackrealign"
+#    export CXX="g++ -m32 -mstackrealign"
+#
+#    echo "slibdir=/usr/lib32" >> configparms
+#    echo "rtlddir=/usr/lib32" >> configparms
+#    echo "sbindir=/usr/bin" >> configparms
+#    echo "rootsbindir=/usr/bin" >> configparms
+#
+#    "${srcdir}"/glibc/configure \
+#        --host=i686-pc-linux-gnu \
+#        --libdir=/usr/lib32 \
+#        --libexecdir=/usr/lib32 \
+#        "${_configure_flags[@]}"
+#
+#    make -O
+#  )
 
   # pregenerate locales here instead of in package
   # functions because localedef does not like fakeroot
@@ -182,6 +183,10 @@ package_glibc() {
   # libraries too. Useful for gdb's catch command.
   install -Dm644 "${srcdir}"/sdt.h "${pkgdir}"/usr/include/sys/sdt.h
   install -Dm644 "${srcdir}"/sdt-config.h "${pkgdir}"/usr/include/sys/sdt-config.h
+
+  # Building with pkgctl we get a segfault, so copy it from the current system
+  # Why static-pie get setfault in virtual-env?
+#  install -Dm755 /usr/bin/ldconfig "${pkgdir}"/usr/bin/ldconfig
 }
 
 package_lib32-glibc() {
