diff --git a/PKGBUILD b/PKGBUILD
index 52ebc94..6c467d8 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -58,8 +58,8 @@ package() {
   cd "$pkgname-$pkgver"
   # project has no install target :(
   install -vDm 755 build/${pkgname}_artefacts/None/Standalone/${_name} -t "${pkgdir}/usr/bin/"
-  install -vDm 755 build/${pkgname}_artefacts/None/VST3/${_name}.vst3/Contents/${CARCH}-linux/${_name}.so \
-    -t "${pkgdir}/usr/lib/vst3/Stochas.vst3/Contents/${CARCH}-linux/"
+  install -vDm 755 build/${pkgname}_artefacts/None/VST3/${_name}.vst3/Contents/`uname -m`-linux/${_name}.so \
+    -t "${pkgdir}/usr/lib/vst3/Stochas.vst3/Contents/`uname -m`-linux/"
 
   install -vDm 644 *.desktop -t "${pkgdir}/usr/share/applications/"
   install -vDm 644 image/app_logo_512.png "${pkgdir}/usr/share/icons/hicolor/512x512/apps/org.surge-synth-team.${_name}.png"
