diff --git a/PKGBUILD b/PKGBUILD
index db2b245..f95fe8a 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -1,14 +1,13 @@
-# Maintainer: Jakub Klinkovský <lahwaacz at archlinux dot org>
 # Maintainer: George Rawlinson <grawlinson@archlinux.org>
 # Maintainer: Sergej Pupykin <pupykin.s+arch@gmail.com>
 
 pkgname=arch-wiki-docs
-pkgver=20240801
-pkgrel=1
+pkgver=20240207
+pkgrel=
 pkgdesc='Pages from Arch Wiki optimized for offline browsing'
 arch=('any')
 url='https://github.com/lahwaacz/arch-wiki-docs'
-license=(GFDL-1.3-or-later)
+license=('FDL')
 makedepends=(
   'git'
   'python'
@@ -17,10 +16,10 @@ makedepends=(
   'python-cssselect'
   'python-requests'
 )
-options=('!strip' '!debug')
-_commit='62059ba9c5ea8589a9277c46fcf069c710f18558'
+options=('!strip')
+_commit='929424410caecd0265e35126f625d9d1a7a63625'
 source=("$pkgname::git+https://github.com/lahwaacz/arch-wiki-docs#commit=$_commit")
-b2sums=('dead6eca758071a06c73c34ad81cd28555245d27d5d2fa21c543559bdece739cca3af5962eecc8c59a9d3d0e9b4444b34a135b915097d594c502f3f7c837f26e')
+b2sums=('SKIP')
 
 pkgver() {
   date "+%Y%m%d"
