diff --git a/PKGBUILD b/PKGBUILD
index 5a4c29e..ab31261 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -17,6 +17,7 @@ groups=('qt5')
 _pkgfqn=${pkgname/5-/}
 source=(git+https://code.qt.io/qt/qtwebengine.git#tag=v${pkgver}-lts
         git+https://code.qt.io/qt/qtwebengine-chromium.git
+        http://loongarch.zhcn.cc/list/qtwebengine/qtwebengine-5.15.2-la64.patch/59168fd4d0e9a62535f8118511fb71fa/qtwebengine-5.15.2-la64.patch
         qt5-webengine-python3.patch
         qt5-webengine-chromium-python3.patch
         qt5-webengine-ffmpeg5.patch
@@ -25,6 +26,7 @@ source=(git+https://code.qt.io/qt/qtwebengine.git#tag=v${pkgver}-lts
         qt5-webengine-icu-74.patch)
 sha256sums=('SKIP'
             'SKIP'
+            'cace5fc41204e2679d577b6dd1030656826bbf69a6a4db126036fad47c7db2d1'
             '0ad5d1660886f7bbf5108b071bf5d7bbbabf1cd1258ce9d4587a01dec4a1aa89'
             'd8beb3d65a1aaf927285e6f055a9d1facd0f9c3fd851f91ba568389fb3137399'
             'c50d3019626183e753c53a997dc8a55938847543aa3178d4c51f377be741c693'
@@ -52,7 +54,7 @@ prepare() {
 
 build() {
   cd build
-  qmake ../${_pkgfqn} CONFIG+=force_debug_info -- \
+  qmake ../${_pkgfqn} CONFIG+=force_debug_info QT_ARCH=loongarch64 -- \
     -proprietary-codecs \
     -system-ffmpeg \
     -webp \
