diff --git a/PKGBUILD b/PKGBUILD
index 65efe0f..40ecc3e 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -16,10 +16,12 @@ groups=('qt5')
 _pkgfqn="qt-everywhere-opensource-src-${pkgver}"
 source=("https://download.qt.io/official_releases/qt/${pkgver%.*}/${pkgver}/single/${_pkgfqn}.tar.xz"
          qt5-webengine-python3.patch
-         no-qmake.patch)
+         no-qmake.patch
+         qt5-base-la64.patch)
 sha256sums=('93f2c0889ee2e9cdf30c170d353c3f829de5f29ba21c119167dee5995e48ccce'
             '398c996cb5b606695ac93645143df39e23fa67e768b09e0da6dbd37342a43f32'
-            'db90fa31381fa0814c9c8c803c9e2f9b36bdd6f52da753399e500c0692352498')
+            'db90fa31381fa0814c9c8c803c9e2f9b36bdd6f52da753399e500c0692352498'
+            '0ea0ced24a71d6abe7d323c4bfb80c33c77592873b8660fe413cb074978c653a')
 
 prepare() {
   cd ${_pkgfqn/opensource-/}
@@ -29,6 +31,7 @@ prepare() {
 
   patch -d qtbase -p1 < "$srcdir"/no-qmake.patch # Use system qmake
   patch -d qtwebengine -p1 < "$srcdir"/qt5-webengine-python3.patch # Fix build with Python 3
+  patch -d qtbase -p1 < "$srcdir"/qt5-base-la64.patch
 }
 
 build() {
