diff --git a/PKGBUILD b/PKGBUILD
index d2e718b..2d2a5f2 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -18,7 +18,7 @@ makedepends=(
   'python-setuptools'
   'python-wheel'
 )
-checkdepends=('python-pytest' 'python-pretend' 'python-flaky')
+makedepends+=('python-pytest' 'python-pretend' 'python-flaky')
 _commit='7f3e4f94701a5e19ec66e3601119dd6d62043cec'
 source=("$pkgname::git+https://github.com/pyca/pyopenssl#commit=$_commit")
 b2sums=('SKIP')
