diff --git a/PKGBUILD b/PKGBUILD
index eee55f5..456910e 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -8,7 +8,7 @@ arch=('any')
 url="https://github.com/amrayn/easyloggingpp"
 license=('MIT')
 makedepends=(cmake ninja)
-checkdepends=(gtest)
+makedepends+=(gtest)
 source=("$pkgname-$pkgver.tar.gz::https://github.com/muflihun/easyloggingpp/archive/v${pkgver}.tar.gz"
         0001-Switch-CMake-build-system-from-C-11-to-C-14.patch)
 sha256sums=('9110638e21ef02428254af8688bf9e766483db8cc2624144aa3c59006907ce22'
