diff --git a/PKGBUILD b/PKGBUILD
index b5ed63d..11f6efb 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -10,20 +10,19 @@ pkgdesc='Lightweight C library that eases the writing of UNIX daemons'
 url='http://0pointer.de/lennart/projects/libdaemon/'
 license=('LGPL')
 arch=('x86_64')
-depends=('glibc')
 makedepends=('git')
-source=(http://0pointer.de/lennart/projects/libdaemon/libdaemon-${pkgver}.tar.gz)
-sha512sums=('a96b25c09bd63cc192c1c5f8b5bf34cc6ad0c32d42ac14b520add611423b6ad3d64091a47e0c7ab9a94476a5e645529abccea3ed6b23596567163fba88131ff2')
+source=('git://git.0pointer.net/libdaemon.git#commit=9fcc28e0e8f84968d1fb8b6d544a42efb13803ec')
+sha512sums=('SKIP')
 
 build() {
-	cd "${srcdir}/${pkgname}-${pkgver}"
+	cd "${srcdir}/${pkgname}"
 	./bootstrap.sh
 	./configure --prefix=/usr --localstatedir=/var --disable-lynx
 	make
 }
 
 package() {
-	cd "${srcdir}/${pkgname}-${pkgver}"
+	cd "${srcdir}/${pkgname}"
 	make DESTDIR="${pkgdir}" install
 }
 
