diff --git a/PKGBUILD b/PKGBUILD
index 249d155..73a34e2 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -24,9 +24,11 @@ makedepends=(
 source=(
   git+https://git.kernel.org/pub/scm/linux/kernel/git/jejb/sbsigntools.git#tag=$_commit?signed
   git+https://git.ozlabs.org/ccan
+  sbsigntools-la64.patch
 )
 sha256sums=('SKIP'
-            'SKIP')
+            'SKIP'
+            '98442f63594d22fc463f5e6dc7d2d6e892e32406ee69a6e5c94e2738e5b3d4ae')
 validpgpkeys=('D5606E73C8B46271BEAD9ADF814AE47C214854D6') # James Bottomley <jejb@kernel.org>
 
 prepare() {
@@ -34,6 +36,7 @@ prepare() {
   git submodule init
   git config submodule."lib/ccan.git".url "$srcdir/ccan"
   git -c protocol.file.allow=always submodule update
+  patch -p1 -i $srcdir/sbsigntools-la64.patch
   ./autogen.sh
 }
 
