diff --git a/PKGBUILD b/PKGBUILD
index 7beae1d..56d5708 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -49,14 +49,14 @@ build() {
 
   # documentation
   local python_version=$(python -c 'import sys; print("".join(map(str, sys.version_info[:2])))')
-  PYTHONPATH="$(pwd)/build/lib.linux-$CARCH-cpython-$python_version:$PYTHONPATH" make docs
+  PYTHONPATH="$(pwd)/build/lib.linux-`uname -m`-cpython-$python_version:$PYTHONPATH" make docs
 }
 
 check() {
   cd "$pkgbase"
 
   local python_version=$(python -c 'import sys; print("".join(map(str, sys.version_info[:2])))')
-  PYTHONPATH="$(pwd)/build/lib.linux-$CARCH-cpython-$python_version:$PYTHONPATH" pytest -v \
+  PYTHONPATH="$(pwd)/build/lib.linux-`uname -m`-cpython-$python_version:$PYTHONPATH" pytest -v \
     --deselect tests/test_event.py::EventTestCase::test_copy \
     --deselect tests/test_event.py::EventTestCase::test_pickle \
     --deselect tests/units/test_modifiers.py::ModifiersTestCase::test_Port
