diff --git a/PKGBUILD b/PKGBUILD
index 92cfbb5..2603332 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -47,10 +47,12 @@ url='https://tug.org/texlive/'
 _commit=871c7a2856d70e1a9703d1f72f0587b9995dba5f # tags/texlive-2023.0
 source=(git+https://github.com/Tex-Live/texlive-source.git#commit=$_commit
         ptex-debug-print.patch
-        context-luatex-1.17.patch)
+        context-luatex-1.17.patch
+        texlive-bin-la64.patch)
 sha256sums=('SKIP'
             'aa838f09003c62c2efb5770a8de66f99b409df049fbd65098d80fd1957d06c50'
-            'a56838d19c3bd820781693b5a2e058e1a22378b37ea199bac426d97fcc420920')
+            'a56838d19c3bd820781693b5a2e058e1a22378b37ea199bac426d97fcc420920'
+            '0c09915ccb8bad1792f11a69cc7057d775cdf54a443427d9ca77f5602432f1aa')
 
 prepare() {
   cd texlive-source
@@ -64,6 +66,7 @@ prepare() {
   patch -p1 -i ../ptex-debug-print.patch
 # update context to work with luatex 1.17
   patch -p1 -i ../context-luatex-1.17.patch
+  patch -p1 -i $srcdir/texlive-bin-la64.patch
 }
 
 build() {
