diff --git a/PKGBUILD b/PKGBUILD
index 6d46b42..a21edc1 100755
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -30,7 +30,7 @@ package() {
   # delete unnecessary files & folders
   cd "$pkgdir/$_gemdir"
   rm -vrf cache
-  rm -vrf "extensions/$CARCH-linux/$(basename $_gemdir)/$_gemname-$pkgver/"{gem_make.out,mkmf.log}
+  rm -vrf "extensions/`uname -m`-linux/$(basename $_gemdir)/$_gemname-$pkgver/"{gem_make.out,mkmf.log}
   cd "gems/$_gemname-$pkgver"
   find . -type f -name ".*" -delete
   rm -vrf Gemfile Rakefile "$_gemname.gemspec" ext test
