diff --git a/PKGBUILD b/PKGBUILD
index 1b70ff6..93da807 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -47,7 +47,7 @@ b2sums=('86cdf27447c003415348705745990899c507b326bc2bc191302cd32f2d6df23b28bfe25
 prepare() {
   cd $pkgbase-$pkgbase-v$pkgver
   # NOTE: add --locked as soon as upstream has split out the components
-  cargo fetch --target "$CARCH-unknown-linux-gnu"
+  cargo fetch --target "`uname -m`-unknown-linux-gnu"
 }
 
 build() {
