diff --git a/PKGBUILD b/PKGBUILD
index a3b3f2c..51e4e6e 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -17,7 +17,7 @@ b2sums=('37e91b5eb41fd56e2be382ee77bd6a6c859d1e1d7c99d45c2597e1a24194ea79ad1c563
 prepare() {
   cd "${pkgname}-${pkgver}"
   sed -i '/\"vendored-libgit2\"/d' Cargo.toml
-  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+  cargo fetch --locked --target "`uname -m`-unknown-linux-gnu"
 }
 
 build() {
