diff --git a/PKGBUILD b/PKGBUILD
index 118ba45..a9b12a8 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -27,7 +27,7 @@ makedepends=(fcft
 optdepends=('libutempter: utmp logging')
 backup=(etc/xdg/foot/foot.ini)
 source=("$pkgname-$pkgver.tar.gz::$url/archive/$pkgver.tar.gz")
-sha256sums=('0e02af376e5f4a96eeb90470b7ad2e79a1d660db2a7d1aa772be43c7db00e475')
+sha256sums=('8060ec28cbf6e2e3d408665330da4bc48fd094d4f1265d7c58dc75c767463c29')
 
 build() {
 	cd "$pkgbase"
