diff --git a/PKGBUILD b/PKGBUILD
index 652aa51..5339b1d 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -26,6 +26,10 @@ prepare() {
 build() {
   cd "${pkgname}" 
   export GOFLAGS="-trimpath -mod=readonly -modcacherw"
+  export GOPROXY=https://goproxy.cn
+  go mod edit -replace=go.etcd.io/bbolt=go.etcd.io/bbolt@v1.3.7-0.20221114114133-eedea6cb26ef
+  go mod edit -replace=github.com/cilium/ebpf=github.com/cilium/ebpf@v0.10.1-0.20230613102335-1140a754d780
+  go mod tidy
   make VERSION=v$pkgver GO_BUILD_FLAGS="-trimpath -mod=readonly -modcacherw" GO_GCFLAGS="" EXTRA_LDFLAGS="-buildid="
   make VERSION=v$pkgver man
 }
