diff --git a/PKGBUILD b/PKGBUILD
index 0b1646e..aeafab3 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -78,7 +78,7 @@ package_openal() {
   install -Dt "$pkgdir/usr/share/doc/openal" -m644 openal-soft/docs/*
 
   cd "$pkgdir"
-  _pick examples usr/bin/al{ffplay,hrtf,latency,loopback,multireverb}
+  _pick examples usr/bin/al{hrtf,latency,loopback,multireverb}
   _pick examples usr/bin/al{play,record,reverb,stream,tonegen}
 }
 
