diff --git a/PKGBUILD b/PKGBUILD
index 85eb5b6..f0da3f0 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -50,7 +50,7 @@ package_libusbsio() {
   )
 
   cd $pkgbase-$pkgver-src
-  install -vDm 755 bin/linux_$CARCH/$pkgbase.so -t "$pkgdir/usr/lib/"
+  install -vDm 755 bin/linux_`uname -m`/$pkgbase.so -t "$pkgdir/usr/lib/"
   install -vDm 644 license/*.txt -t "$pkgdir/usr/share/licenses/$pkgname/"
 }
 
@@ -67,8 +67,8 @@ package_python-libusbsio() {
   (
     cd python/dist/$pkgbase-$pkgver
     python -m installer --destdir="$pkgdir" dist/*.whl
-    install -vdm 755 "$pkgdir/$_site_packages/$pkgbase/bin/linux_$CARCH"
-    ln -fsv /usr/lib/$pkgbase.so "$pkgdir/$_site_packages/$pkgbase/bin/linux_$CARCH/$pkgbase.so"
+    install -vdm 755 "$pkgdir/$_site_packages/$pkgbase/bin/linux_`uname -m`"
+    ln -fsv /usr/lib/$pkgbase.so "$pkgdir/$_site_packages/$pkgbase/bin/linux_`uname -m`/$pkgbase.so"
   )
   install -vDm 644 license/BSD-3-clause.txt -t "$pkgdir/usr/share/licenses/$pkgname/"
   install -vDm 644 python/README.md -t "$pkgdir/usr/share/doc/$pkgname/"
