diff --git a/PKGBUILD b/PKGBUILD
index 1a7172a..1ca2513 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -24,6 +24,10 @@ sha256sums=('f885e17998dc1eece1688a75e516663462fe72a7f4f132def4132055777c7ff8'
 
 build() {
   cd fossil-src-$pkgver
+  if [ -f autosetup/autosetup-config.guess ]; then
+    cp /usr/share/automake-1.16/config.guess autosetup/autosetup-config.guess
+    cp /usr/share/automake-1.16/config.sub autosetup/autosetup-config.sub
+  fi
   ./configure --prefix=/usr --json --disable-internal-sqlite --with-tcl=/usr --with-tcl-private-stubs=1
   # headers and translate targets are problematic with parallel jobs
   #make -j1 bld bld/headers
