diff --git a/PKGBUILD b/PKGBUILD
index 34d0d4a..d8fef88 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -19,7 +19,7 @@ makedepends=(
   cmake
   ninja
   nasm
-  'java-environment>11'
+  'jdk11-openjdk'
 )
 optdepends=('java-runtime>11: for TurboJPEG Java wrapper')
 provides=(
@@ -45,6 +45,7 @@ build() {
     -D ENABLE_STATIC=OFF
     -D WITH_JAVA=ON
     -D WITH_JPEG8=ON
+    -D WITH_SIMD=OFF
     -G Ninja
     -S $pkgname-$pkgver
     -W no-dev
