diff --git a/PKGBUILD b/PKGBUILD
index 60f19d3..b5d76c6 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -26,14 +26,17 @@ _base_pkgver=$_pkgver_major.$_pkgver_minor.x
 
 source=(http://ftp.rpm.org/releases/rpm-$_base_pkgver/rpm-$pkgver.tar.bz2
 	rpmextract.sh
-        rpmlib-filesystem-check.patch)
+        rpmlib-filesystem-check.patch
+        rpm-add-loongarch.patch)
 sha256sums=('ba7eee1bc2c6f83be73c0a40d159c625cbaed976b3ac044233404fb25ae1b979'
             '3e5bf450d4628366ba35469ec0530a99cd09ab2616a3d261a3f68270f481f777'
-            'bd0e6dbd458f990268c60324190c6825b234647ecdde08296d2b453dc4bce27a')
+            'bd0e6dbd458f990268c60324190c6825b234647ecdde08296d2b453dc4bce27a'
+            'a40afdd567b77480d2e315d3ca297f5238dd3f9f11cccc239a1d33b297b84e04')
 
 prepare() {
 	cd rpm-${pkgver}
 	patch -p1 < ../rpmlib-filesystem-check.patch
+    patch -p1 -i $srcdir/rpm-add-loongarch.patch
 }
 
 build() {
