diff --git a/PKGBUILD b/PKGBUILD
index 9c3080a..ec88695 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -6,7 +6,7 @@ pkgname=phosh
 pkgver=0.35.0
 pkgrel=1
 pkgdesc='A pure Wayland shell prototype for GNOME on mobile devices'
-arch=(x86_64 aarch64)
+arch=(loong64 x86_64 aarch64)
 url='https://gitlab.gnome.org/World/Phosh/phosh'
 license=(GPL3)
 depends=(gtk3 libhandy gnome-desktop gnome-session upower libpulse gcr
