diff --git a/PKGBUILD b/PKGBUILD
index 8360481..b8ec8c8 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -14,6 +14,12 @@ sha512sums=('c0a51e03293286498923d0d0666950f0da140f1347d9953751102bff5b01b47c64f
 
 build() {
   cd "${pkgname}-${pkgver}"
+  find -name Cargo.lock -exec rm -f {} \;
+  mkdir -p .cargo
+  cat > .cargo/config.toml <<EOF
+[source.crates-io]
+registry = "https://gitee.com/yetist/crates.io-index"
+EOF
   cargo build --release --all-features --target-dir=target
 }
 
