diff --git a/PKGBUILD b/PKGBUILD
index 1ebbf05..756ddc3 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -63,9 +63,12 @@ build() {
     --disable-static
     --enable-glib
     --prefix=/usr
+    --enable-glib --disable-examples --disable-tests
   )
 
   cd $_pkgname-$pkgver
+  # -lpthread needed
+  export LDFLAGS+=" -lpthread"
   ./configure "${configure_options[@]}"
   # prevent excessive overlinking due to libtool
   sed -i -e 's/ -shared / -Wl,-O1,--as-needed\0/g' libtool
