diff --git a/PKGBUILD b/PKGBUILD
index 81240b9..21f45b8 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -21,7 +21,7 @@ b2sums=('c1907710867b3c1342cc5c9661a095c5c4f62c52a8284eccba8ff71398d933667924df6
 prepare() {
   cd $_upstream_name-$pkgname-$pkgver
   export RUSTUP_TOOLCHAIN=stable
-  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+  cargo fetch --locked --target "`uname -m`-unknown-linux-gnu"
 }
 
 build() {
