diff --git a/PKGBUILD b/PKGBUILD
index 0dad8b7..cf0971b 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -20,7 +20,7 @@ makedepends=(
   python-setuptools
   python-wheel
 )
-checkdepends=(
+makedepends+=(
   boost
   clang
   cmake
@@ -37,18 +37,18 @@ checkdepends=(
   graphviz
   gtest
   gtk-doc
-  gtk-sharp-2
+#  gtk-sharp-2
   gtk3
   gtkmm3
   hotdoc
   itstool
   java-environment=8
-  ldc
+#  ldc
   libelf
   libwmf
   llvm
   mercurial
-  mono
+#  mono
   nasm
   netcdf-fortran
   openmpi
@@ -58,8 +58,8 @@ checkdepends=(
   python-pytest-xdist
   qt5-base
   qt5-tools
-  rust
-  rust-bindgen
+#  rust
+#  rust-bindgen
   sdl2
   vala
   valgrind
