diff --git a/PKGBUILD b/PKGBUILD
index a90b392..a1285cc 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -87,7 +87,7 @@ check() {
   cd ${pkgname}
   local _python_version=$(python -c 'import sys; print("".join(map(str, sys.version_info[:2])))')
   mv tests/autobahn/test_autobahn.py{,.bak} # Docker tests
-  PYTHONPATH="$PWD/build/lib.linux-$CARCH-cpython-${_python_version}" pytest
+  PYTHONPATH="$PWD/build/lib.linux-`uname -m`-cpython-${_python_version}" pytest
   mv tests/autobahn/test_autobahn.py{.bak,}
 }
 
