diff --git a/PKGBUILD b/PKGBUILD
index 213d589..f301b52 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -14,6 +14,12 @@ sha256sums=('ff1a34d3df9a1a5e5c1fa3895c036a885dc7b9740d7fccdf57e9ed678b8fb3a3')
 
 build() {
     cd "$srcdir/$pkgname-$pkgver/imagequant-sys"
+    find -name Cargo.lock -exec rm -f {} \;
+    mkdir -p .cargo
+    cat > .cargo/config.toml <<EOF
+[source.crates-io]
+registry = "https://gitee.com/yetist/crates.io-index"
+EOF
     cargo cbuild --release --prefix=/usr
 }
 
